<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>FARM: simulating the spread of agriculture through cultural phylogenies</title>
  <meta name="description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook.">
  <meta name="generator" content="bookdown 0.5 and GitBook 2.6.7">

  <meta property="og:title" content="FARM: simulating the spread of agriculture through cultural phylogenies" />
  <meta property="og:type" content="book" />
  
  <meta property="og:image" content="Conffusion_matrix_all.png" />
  <meta property="og:description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="FARM: simulating the spread of agriculture through cultural phylogenies" />
  
  <meta name="twitter:description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  <meta name="twitter:image" content="Conffusion_matrix_all.png" />

<meta name="author" content="Ty Tuff, Bruno Vilela, and Carlos Botero">



  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="module-1-simulation-to-produce-a-world-and-a-tree.html">
<link rel="next" href="calling-module-1-and-module-2.html">
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />









<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">A Minimal Book Example</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="spatial-scaffolding-for-the-simulations.html"><a href="spatial-scaffolding-for-the-simulations.html"><i class="fa fa-check"></i><b>2</b> Spatial scaffolding for the simulations</a></li>
<li class="chapter" data-level="3" data-path="rule-sets.html"><a href="rule-sets.html"><i class="fa fa-check"></i><b>3</b> Rule sets</a><ul>
<li class="chapter" data-level="3.1" data-path="rule-sets.html"><a href="rule-sets.html#simulation"><i class="fa fa-check"></i><b>3.1</b> Simulation:</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="module-1-simulation-to-produce-a-world-and-a-tree.html"><a href="module-1-simulation-to-produce-a-world-and-a-tree.html"><i class="fa fa-check"></i><b>4</b> Module 1: Simulation to produce a world and a tree</a><ul>
<li class="chapter" data-level="4.1" data-path="module-1-simulation-to-produce-a-world-and-a-tree.html"><a href="module-1-simulation-to-produce-a-world-and-a-tree.html#inputs"><i class="fa fa-check"></i><b>4.1</b> Inputs</a></li>
<li class="chapter" data-level="4.2" data-path="module-1-simulation-to-produce-a-world-and-a-tree.html"><a href="module-1-simulation-to-produce-a-world-and-a-tree.html#module-1-functions"><i class="fa fa-check"></i><b>4.2</b> Module 1 functions</a></li>
<li class="chapter" data-level="4.3" data-path="module-1-simulation-to-produce-a-world-and-a-tree.html"><a href="module-1-simulation-to-produce-a-world-and-a-tree.html#push-versions"><i class="fa fa-check"></i><b>4.3</b> Push versions</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="module-2-space-and-phylogeny-summary-statistics.html"><a href="module-2-space-and-phylogeny-summary-statistics.html"><i class="fa fa-check"></i><b>5</b> Module 2: Space and phylogeny summary statistics</a><ul>
<li class="chapter" data-level="5.1" data-path="module-2-space-and-phylogeny-summary-statistics.html"><a href="module-2-space-and-phylogeny-summary-statistics.html#phylogenetic-summary-statistics"><i class="fa fa-check"></i><b>5.1</b> Phylogenetic summary statistics</a><ul>
<li class="chapter" data-level="5.1.1" data-path="module-2-space-and-phylogeny-summary-statistics.html"><a href="module-2-space-and-phylogeny-summary-statistics.html#introduction-and-framework"><i class="fa fa-check"></i><b>5.1.1</b> Introduction and framework</a></li>
<li class="chapter" data-level="5.1.2" data-path="module-2-space-and-phylogeny-summary-statistics.html"><a href="module-2-space-and-phylogeny-summary-statistics.html#alpha-diversity-metrics"><i class="fa fa-check"></i><b>5.1.2</b> Alpha diversity metrics</a></li>
<li class="chapter" data-level="5.1.3" data-path="module-2-space-and-phylogeny-summary-statistics.html"><a href="module-2-space-and-phylogeny-summary-statistics.html#pairwise-distance-between-tips"><i class="fa fa-check"></i><b>5.1.3</b> Pairwise distance between tips</a></li>
<li class="chapter" data-level="5.1.4" data-path="module-2-space-and-phylogeny-summary-statistics.html"><a href="module-2-space-and-phylogeny-summary-statistics.html#nearest-phylogenetic-neighbor"><i class="fa fa-check"></i><b>5.1.4</b> Nearest phylogenetic neighbor</a></li>
<li class="chapter" data-level="5.1.5" data-path="module-2-space-and-phylogeny-summary-statistics.html"><a href="module-2-space-and-phylogeny-summary-statistics.html#phylogenetic-isolation"><i class="fa fa-check"></i><b>5.1.5</b> Phylogenetic isolation</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="module-2-space-and-phylogeny-summary-statistics.html"><a href="module-2-space-and-phylogeny-summary-statistics.html#beta-diversity"><i class="fa fa-check"></i><b>5.2</b> Beta diversity</a></li>
<li class="chapter" data-level="5.3" data-path="module-2-space-and-phylogeny-summary-statistics.html"><a href="module-2-space-and-phylogeny-summary-statistics.html#tree-topology"><i class="fa fa-check"></i><b>5.3</b> Tree topology</a></li>
<li class="chapter" data-level="5.4" data-path="module-2-space-and-phylogeny-summary-statistics.html"><a href="module-2-space-and-phylogeny-summary-statistics.html#macroevolutionary-rates"><i class="fa fa-check"></i><b>5.4</b> Macroevolutionary rates</a></li>
<li class="chapter" data-level="5.5" data-path="module-2-space-and-phylogeny-summary-statistics.html"><a href="module-2-space-and-phylogeny-summary-statistics.html#spatial-locations"><i class="fa fa-check"></i><b>5.5</b> Spatial Locations</a></li>
<li class="chapter" data-level="5.6" data-path="module-2-space-and-phylogeny-summary-statistics.html"><a href="module-2-space-and-phylogeny-summary-statistics.html#module2-returns-these-two-matrices-as-a-list"><i class="fa fa-check"></i><b>5.6</b> Module2() returns these two matrices as a list</a><ul>
<li class="chapter" data-level="5.6.1" data-path="module-2-space-and-phylogeny-summary-statistics.html"><a href="module-2-space-and-phylogeny-summary-statistics.html#here-is-the-exact-version-in-r"><i class="fa fa-check"></i><b>5.6.1</b> Here is the exact version in R</a></li>
</ul></li>
<li class="chapter" data-level="5.7" data-path="module-2-space-and-phylogeny-summary-statistics.html"><a href="module-2-space-and-phylogeny-summary-statistics.html#references"><i class="fa fa-check"></i><b>5.7</b> References</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="calling-module-1-and-module-2.html"><a href="calling-module-1-and-module-2.html"><i class="fa fa-check"></i><b>6</b> Calling Module 1 and Module 2</a><ul>
<li class="chapter" data-level="6.1" data-path="calling-module-1-and-module-2.html"><a href="calling-module-1-and-module-2.html#run-the-simulation-to-a-specified-timestep-and-then-save-one-output"><i class="fa fa-check"></i><b>6.1</b> Run the simulation to a specified timestep and then save one output</a></li>
<li class="chapter" data-level="6.2" data-path="calling-module-1-and-module-2.html"><a href="calling-module-1-and-module-2.html#run-simulations-for-a-specified-time-but-run-stats-and-save-timesteps-along-the-way"><i class="fa fa-check"></i><b>6.2</b> Run simulations for a specified time but run stats and save timesteps along the way</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="running-module-1-and-module-2-on-the-cluster.html"><a href="running-module-1-and-module-2-on-the-cluster.html"><i class="fa fa-check"></i><b>7</b> Running Module 1 and Module 2 on the cluster</a><ul>
<li class="chapter" data-level="7.1" data-path="running-module-1-and-module-2-on-the-cluster.html"><a href="running-module-1-and-module-2-on-the-cluster.html#not-for-the-faint-of-hart"><i class="fa fa-check"></i><b>7.1</b> Not for the faint of hart</a></li>
<li class="chapter" data-level="7.2" data-path="running-module-1-and-module-2-on-the-cluster.html"><a href="running-module-1-and-module-2-on-the-cluster.html#bash-scripts"><i class="fa fa-check"></i><b>7.2</b> Bash scripts</a></li>
<li class="chapter" data-level="7.3" data-path="running-module-1-and-module-2-on-the-cluster.html"><a href="running-module-1-and-module-2-on-the-cluster.html#passing-arguements-to-r-script"><i class="fa fa-check"></i><b>7.3</b> Passing arguements to R-script</a></li>
<li class="chapter" data-level="7.4" data-path="running-module-1-and-module-2-on-the-cluster.html"><a href="running-module-1-and-module-2-on-the-cluster.html#accessing-and-working-with-the-cluster"><i class="fa fa-check"></i><b>7.4</b> Accessing and working with the cluster</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="module-3-analysing-results-produce-from-modules-1-and-2.html"><a href="module-3-analysing-results-produce-from-modules-1-and-2.html"><i class="fa fa-check"></i><b>8</b> Module 3: Analysing results produce from Modules 1 and 2</a><ul>
<li class="chapter" data-level="8.1" data-path="module-3-analysing-results-produce-from-modules-1-and-2.html"><a href="module-3-analysing-results-produce-from-modules-1-and-2.html#organize-data"><i class="fa fa-check"></i><b>8.1</b> Organize data</a></li>
<li class="chapter" data-level="8.2" data-path="module-3-analysing-results-produce-from-modules-1-and-2.html"><a href="module-3-analysing-results-produce-from-modules-1-and-2.html#diagnostics"><i class="fa fa-check"></i><b>8.2</b> Diagnostics</a></li>
<li class="chapter" data-level="8.3" data-path="module-3-analysing-results-produce-from-modules-1-and-2.html"><a href="module-3-analysing-results-produce-from-modules-1-and-2.html#random-forest-analysis"><i class="fa fa-check"></i><b>8.3</b> Random Forest Analysis</a><ul>
<li class="chapter" data-level="8.3.1" data-path="module-3-analysing-results-produce-from-modules-1-and-2.html"><a href="module-3-analysing-results-produce-from-modules-1-and-2.html#trainingbuilding"><i class="fa fa-check"></i><b>8.3.1</b> Training/Building</a></li>
<li class="chapter" data-level="8.3.2" data-path="module-3-analysing-results-produce-from-modules-1-and-2.html"><a href="module-3-analysing-results-produce-from-modules-1-and-2.html#testing"><i class="fa fa-check"></i><b>8.3.2</b> Testing</a></li>
<li class="chapter" data-level="8.3.3" data-path="module-3-analysing-results-produce-from-modules-1-and-2.html"><a href="module-3-analysing-results-produce-from-modules-1-and-2.html#matching"><i class="fa fa-check"></i><b>8.3.3</b> Matching</a></li>
</ul></li>
<li class="chapter" data-level="8.4" data-path="module-3-analysing-results-produce-from-modules-1-and-2.html"><a href="module-3-analysing-results-produce-from-modules-1-and-2.html#run-a-single-random-forest-on-available-outputs"><i class="fa fa-check"></i><b>8.4</b> Run a single random forest on available outputs</a></li>
<li class="chapter" data-level="8.5" data-path="module-3-analysing-results-produce-from-modules-1-and-2.html"><a href="module-3-analysing-results-produce-from-modules-1-and-2.html#visualize-outputs-from-random-forest-analysis"><i class="fa fa-check"></i><b>8.5</b> Visualize outputs from Random Forest analysis</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">FARM: simulating the spread of agriculture through cultural phylogenies</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="module-2-space-and-phylogeny-summary-statistics" class="section level1">
<h1><span class="header-section-number">Chapter 5</span> Module 2: Space and phylogeny summary statistics</h1>
<p>This is the master document for Module 2, a foundational function in our FARM package that analyzes results from Module 1, the other foundational function. Module 1 simulates a spatial pattern and a phylogenetic tree given a set of environmental and inheritance rules and then Module 2 summarizes those simulated results using a large set of targeted summary statistics. Here we describe our choice of summary statistics, justify those choices as part of a larger theoretical context, and provide our reproducable code for executing the anaylses yourself. These two parts are seperated into modules so that they can act independently. An combination of spatial pattern and associated phylogeny many be used as long as they are formatted correctly.</p>
<p>This pipeline was designed to analyze a simulated world where all the information is known about both the world and the tree. There is no missing information, just extinct trees. This is much different than our real tree that has loads of uncertainty unevenly destributed across it. The result you see demonstrated right now are one simulated result of many. I need to do a sister page to this were we do this entire analysis on the real tree, or best real tree we’ve got.</p>
<p>We have four types of data available for asking research questions using D-place data: <a href="module-2-space-and-phylogeny-summary-statistics.html#phylogenetic-summary-statistics">phylogenies</a>, <a href="module-2-space-and-phylogeny-summary-statistics.html#spatial-locations">spatial locations</a>, trait identities, and environmental reconstructions. Any one of these four data types alone are relatively information poor, so we are searching for ways to model connections between these data types to draw stronger conclusions overall.</p>
<p>Other modules can use the summary statistics generated from this module to test hypotheses. We currently have a ABC and Random Forest module started but there will be more to come.</p>
<p>These are quantitative connections that we are assumed in the analyses, but we don’t actually have any support in the data for doing so. 1. nearest neighbor connectivity measures 2. Abundance estimates 3. Pairwise influence (history) between cultures. 4. Environmental reconstruction validation evidence</p>
<div id="phylogenetic-summary-statistics" class="section level2">
<h2><span class="header-section-number">5.1</span> Phylogenetic summary statistics</h2>
<p>Whole tree vs. part of tree? These statistics are generally used to compare one sample to another. For example, an experimental contrast between two sites, two phylogenetic groups, or two communities in two different locations. Here we are calculating these statistics for the global langauage tree to compare against global trees created in our simulation. You still retain the ability to subset this tree or others and send only those subsets through this code to compare the values with each other afterwards.</p>
<ul>
<li>Introduction and framework</li>
<li><a href="module-2-space-and-phylogeny-summary-statistics.html#alpha-diversity-metrics">Alpha Diversity metrics</a></li>
</ul>
<ol style="list-style-type: decimal">
<li><a href="module-2-space-and-phylogeny-summary-statistics.html#branch-lengths">Branch Length (richness and divergence)</a></li>
<li><a href="module-2-space-and-phylogeny-summary-statistics.html#pairwise-distance-between-tips">Pairwise distance between tips (richness, divergence, and regularity)</a></li>
<li><a href="module-2-space-and-phylogeny-summary-statistics.html#phylogenetic-isolation">Phylogenetic isolation (divergence, and regularity)</a></li>
<li><a href="module-2-space-and-phylogeny-summary-statistics.html#nearest-phylogenetic-neighbor">Nearest Neighbor (divergence, and regularity)</a></li>
</ol>
<ul>
<li><a href="module-2-space-and-phylogeny-summary-statistics.html#beta-diversity">Beta Diversity</a></li>
<li><a href="module-2-space-and-phylogeny-summary-statistics.html#tree-topology">Tree topology</a></li>
<li><a href="module-2-space-and-phylogeny-summary-statistics.html#macroevolutionary-rates">Macroevolutionary rates</a></li>
</ul>
<p>All trees are ultrametric.</p>
<div id="introduction-and-framework" class="section level3">
<h3><span class="header-section-number">5.1.1</span> Introduction and framework</h3>
<p>The choice of phylogenetic analyses and organizational scheme is based on the suggestions of <span class="citation">Tucker et al. (<a href="#ref-Tucker2016">2016</a>)</span>. Here are a few images from that paper for an overview:</p>
<p><img src="Images/Tucker_2016/tree_concept.jpg" alt="From Tucker et al. (2016)" /> <img src="Images/Tucker_2016/alpha_PCA.jpg" alt="From Tucker et al. (2016)" /></p>
<div class="figure">
<img src="Images/Tucker_2016/Three_type_summary.jpg" alt="From Tucker et al. (2016)" />
<p class="caption">From <span class="citation">Tucker et al. (<a href="#ref-Tucker2016">2016</a>)</span></p>
</div>
<div class="figure">
<img src="Images/Tucker_2016/tutorial.jpg" alt="From Tucker et al. (2016)" />
<p class="caption">From <span class="citation">Tucker et al. (<a href="#ref-Tucker2016">2016</a>)</span></p>
</div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(knitr)
<span class="kw">library</span>(phytools)</code></pre></div>
<pre><code>## Loading required package: ape</code></pre>
<pre><code>## 
## Attaching package: &#39;ape&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:spdep&#39;:
## 
##     plot.mst</code></pre>
<pre><code>## 
## Attaching package: &#39;phytools&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:Matrix&#39;:
## 
##     expm</code></pre>
<pre><code>## The following object is masked from &#39;package:FARM&#39;:
## 
##     bd</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(FARM)
<span class="kw">library</span>(ROCR)</code></pre></div>
<pre><code>## Loading required package: gplots</code></pre>
<pre><code>## 
## Attaching package: &#39;gplots&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:stats&#39;:
## 
##     lowess</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(spdep)</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">load</span>(<span class="st">&#39;~/Downloads/download.Rdata&#39;</span>)

this_tree &lt;-<span class="st"> </span>myOut<span class="op">$</span>mytree
this_world &lt;-<span class="st"> </span>myOut<span class="op">$</span>myWorld</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">str</span>(this_world)</code></pre></div>
<pre><code>## &#39;data.frame&#39;:    1253 obs. of  8 variables:
##  $ cellID     : num  1 2 3 4 5 6 7 8 9 10 ...
##  $ Longitude  : num  -60 12 28 -124 -63 ...
##  $ Latitude   : num  -25 10 -29 54 4 ...
##  $ Parent     : num  828 NA 216 616 901 NA NA NA NA NA ...
##  $ BirthT     : num  37.8 NA 37.6 31 36.2 ...
##  $ Trait      : num  1 NA 1 1 1 NA NA NA NA NA ...
##  $ Environment: num  2 2 2 1 2 1 2 1 1 1 ...
##  $ TipLabel   : chr  &quot;t1&quot; &quot;t2&quot; &quot;t3&quot; &quot;t4&quot; ...</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">str</span>(this_tree)</code></pre></div>
<pre><code>## List of 4
##  $ edge       : num [1:910, 1:2] 457 460 463 560 892 892 560 463 466 466 ...
##  $ tip.label  : chr [1:456] &quot;t862&quot; &quot;t409&quot; &quot;t260&quot; &quot;t204&quot; ...
##  $ edge.length: num [1:910] 2.667 4.433 20.973 10.011 0.916 ...
##  $ Nnode      : num 455
##  - attr(*, &quot;class&quot;)= chr &quot;phylo&quot;
##  - attr(*, &quot;order&quot;)= chr &quot;cladewise&quot;</code></pre>
</div>
<div id="alpha-diversity-metrics" class="section level3">
<h3><span class="header-section-number">5.1.2</span> Alpha diversity metrics</h3>
<div id="branch-lengths" class="section level4">
<h4><span class="header-section-number">5.1.2.1</span> Branch Lengths</h4>
<p>Branch length data is embedded in the tree object provided to this function. The first step in summarizing the lengths is to extract those data from the tree object. These data are called ‘edges’ in the tree object. We extract branch lengths and create an object called ‘Branch_lengths’ for passing on to the other summary functions. The histogram below shows the frequency of different branch lengths found throughout the tree.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">Branch_Lengths &lt;-<span class="st"> </span>this_tree<span class="op">$</span>edge.length</code></pre></div>
<p><img src="bookdown-demo_files/figure-html/unnamed-chunk-19-1.png" width="672" /> We can summarize branch lengths according to normal summary statistics, but it can be difficult to assign evolutionary meaning to some of these metrics and so they are not regularly used as best I can tell. This lack of meaning does not mean that these statistics couldn’t be used to distinguish between large simulated trees.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">mean_branch_length &lt;-<span class="st"> </span><span class="kw">mean</span>(Branch_Lengths)
variance_branch_length &lt;-<span class="st"> </span><span class="kw">var</span>(Branch_Lengths)
SD_branch_length &lt;-<span class="st"> </span><span class="kw">sd</span>(Branch_Lengths)</code></pre></div>
<pre><code>## [1] &quot;mean branch length =  3.64753764473253&quot;</code></pre>
<pre><code>## [1] &quot;variance in branch lengths =  14.9947769344804&quot;</code></pre>
<pre><code>## [1] &quot;standard deviation in branch lengths =  3.87230899263998&quot;</code></pre>
<p><img src="bookdown-demo_files/figure-html/unnamed-chunk-22-1.png" width="672" /></p>
</div>
<div id="phylogenetic-diversity-pd" class="section level4">
<h4><span class="header-section-number">5.1.2.2</span> Phylogenetic diversity (<span class="math inline">\(PD\)</span>)</h4>
<p>Phylogenetic diversity (<span class="math inline">\(PD\)</span>) is the summation (<span class="math inline">\(\sum\)</span>) of all branch lengths connecting species together, where <span class="math inline">\(B_{t}\)</span> is the set of included tips and <span class="math inline">\(L_{b}\)</span> is Branch lengths <span class="citation">(Faith <a href="#ref-Faith1992">1992</a>)</span>. This is an anchor test, which means it is regularly used, well understood, and we should use it to anchor our work to past work. PD is a richness measure, it tells us how much evolutionary history is associated with a set of tips.</p>
<p><span class="math display">\[PD = \sum_{b \in B_{t}}^{}L_{b}\]</span></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Anchor test = PD (Faith&#39;s phylogenetic diversity)</span>
Pylo_diversity_is_sum_of_BL &lt;-<span class="st"> </span><span class="kw">sum</span>(Branch_Lengths)
Pylo_diversity_is_sum_of_BL</code></pre></div>
<pre><code>## [1] 3319.259</code></pre>
<p>There are variations on this measure that we have NOT implemented here. It is popular to scale this measure according to some ecological driver. <span class="citation">Barker (<a href="#ref-Barker2002">2002</a>)</span> scales branch lengths (<span class="math inline">\(L_{b}\)</span>) by multiplying them against the abundance of individuals at at tip (<span class="math inline">\(A_{b}\)</span>). Others <span class="citation">(D. F. Rosauer et al. <a href="#ref-Rosauer2009">2009</a>)</span>, scale them by their range size instead (<span class="math inline">\(R_{b}\)</span>).</p>
<p><span class="math display">\[\Delta n PD = \sum_{b \in B_{t}}^{}A_{b}L_{b}\]</span> <span class="math display">\[PE = \sum_{b \in B_{t}}^{}\dfrac{L_{b}}{R_{b}}\]</span> Argueing that proportional abundance phylogenetic diversity (<span class="math inline">\(PD_{Ab}\)</span>) is more effective than the standard PD calculated from raw abundance, <span class="citation">Vellend et al. (<a href="#ref-Vellend2011">2011</a>)</span> penned a new version of PD where <span class="math inline">\(B\)</span> is the total number of branch lengths (<span class="math inline">\(L_{b}\)</span>). Note: We don’t have abundance data right now for the human project so this metric is not currently very helpful.<br />
<span class="math display">\[PD_{Ab} = B * \dfrac{\sum_{b \in B_{t}}^{}A_{b}L_{b}}{\sum_{b \in B_{t}}^{}A_{b}}\]</span></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#Calculate B</span>
number_of_branches &lt;-<span class="st"> </span><span class="kw">length</span>(Branch_Lengths)
number_of_branches</code></pre></div>
<pre><code>## [1] 910</code></pre>
</div>
<div id="average-phylogenetic-diversity-avpd" class="section level4">
<h4><span class="header-section-number">5.1.2.3</span> Average phylogenetic diversity (<span class="math inline">\(avPD\)</span>)</h4>
<p>Average phylogenetic diversity (<span class="math inline">\(avPD\)</span>) <span class="citation">(Clarke and Warwick <a href="#ref-Clarke2001">2001</a>)</span> is a branch length-based divergence indices where PD is divided by the total number of tips (<span class="math inline">\(S\)</span>) in the tree. <span class="math display">\[avPD = \dfrac{PD}{S}\]</span></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">Number_of_tips &lt;-<span class="st"> </span><span class="kw">length</span>(this_tree<span class="op">$</span>tip.label)
average_phylogenetic_diversity &lt;-<span class="st"> </span>Pylo_diversity_is_sum_of_BL <span class="op">/</span><span class="st"> </span>Number_of_tips
average_phylogenetic_diversity</code></pre></div>
<pre><code>## [1] 7.279077</code></pre>
<p>There is also a proportional abundance version of average phylogenetic diversity (<span class="math inline">\(avPD_{Ab}\)</span>) <span class="citation">(Tucker et al. <a href="#ref-Tucker2016">2016</a>)</span>. Again, we don’t have abundance values yet for D-place. <span class="math display">\[avPD_{Ab} = \dfrac{B * \dfrac{\sum_{b \in B_{t}}^{}A_{b}L_{b}}{\sum_{b \in B_{t}}^{}A_{b}}}{S}\]</span></p>
</div>
</div>
<div id="pairwise-distance-between-tips" class="section level3">
<h3><span class="header-section-number">5.1.3</span> Pairwise distance between tips</h3>
<p>This is the patristic distance, the sum of the branch lengths following the shortest distance between two tips in a tree, implemented as a distance matrix where every tip is compared to every other tip. This distance function can be anything. We use euclidean and environmental distance matrices heavily in the spatial analyses.</p>
<div id="calculate-the-patristic-distance-between-two-taxa-for-all-taxa" class="section level4">
<h4><span class="header-section-number">5.1.3.1</span> Calculate the patristic distance between two taxa, for all taxa</h4>
<p>Calculate the patristic distance between two taxa using the R package ‘phytools’, this function takes a ‘phylo’ tree object and returns a distance matrix between tips. Need original citation.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">## Pairwise distance between tips - From library(ape) in library(phytools)
Pairwise_dist &lt;-<span class="st"> </span><span class="kw">cophenetic</span>(this_tree)</code></pre></div>
<p>yields a distance matrix (list of 2D matrices) of all distances between taxa.</p>
<pre><code>##  num [1:456, 1:456] 0 57.9 78 57.9 72.7 ...
##  - attr(*, &quot;dimnames&quot;)=List of 2
##   ..$ : chr [1:456] &quot;t862&quot; &quot;t409&quot; &quot;t260&quot; &quot;t204&quot; ...
##   ..$ : chr [1:456] &quot;t862&quot; &quot;t409&quot; &quot;t260&quot; &quot;t204&quot; ...</code></pre>
</div>
<div id="sum-of-all-pairwise-distances-f" class="section level4">
<h4><span class="header-section-number">5.1.3.2</span> Sum of all pairwise distances (<span class="math inline">\(F\)</span>)</h4>
<p>Now we can use a set of summary statistics to describe those pairwise distances. The sum of all pairwise distances, <span class="math inline">\(F\)</span>, is formally called ‘Extensive quadratic entropy’. <span class="citation">(Izsak and Papp <a href="#ref-Izsak2000">2000</a>)</span>. Just as it was with branch lengths, this is a richness measure and, accordingly, should be used to answer richness questions.</p>
<p><span class="math display">\[F = \sum_{i} \sum_{j} d_{ij}\]</span></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># F -- Extensive quadratic entropy</span>
F_quadratic_entropy_is_sum_of_PD &lt;-<span class="st"> </span><span class="kw">sum</span>(Pairwise_dist)
F_quadratic_entropy_is_sum_of_PD</code></pre></div>
<pre><code>## [1] 14047351</code></pre>
</div>
<div id="mean-pairwise-distance-mpd" class="section level4">
<h4><span class="header-section-number">5.1.3.3</span> Mean pairwise distance (MPD)</h4>
<p>Mean inter-species distances. The mean of all pairwise distances, <span class="math inline">\(MPD\)</span> (a.k.a. <span class="math inline">\(AvTD\)</span>, and <span class="math inline">\(\Delta^{+}\)</span>), is the mean distance between species. <span class="citation">(Clarke and Warwick <a href="#ref-Clarke1998">1998</a>; Webb et al. <a href="#ref-Webb2002">2002</a>; Webb, Ackerly, and Kembel <a href="#ref-Webb2008">2008</a>; Kembel et al. <a href="#ref-Kembel2010">2010</a>)</span>. <span class="math display">\[MPD = \dfrac{\sum_{ij} d_{ij}}{S(S-1)}\]</span></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Anchor test = MPD (mean pairwise distance)</span>
Mean_pairwise_distance &lt;-<span class="st"> </span>
<span class="st">  </span>Pairwise_dist <span class="op">/</span><span class="st"> </span>(Number_of_tips <span class="op">*</span><span class="st"> </span>(Number_of_tips <span class="op">-</span><span class="st"> </span><span class="dv">1</span>) ) </code></pre></div>
<pre><code>##  num [1:456, 1:456] 0 0.000279 0.000376 0.000279 0.00035 ...
##  - attr(*, &quot;dimnames&quot;)=List of 2
##   ..$ : chr [1:456] &quot;t862&quot; &quot;t409&quot; &quot;t260&quot; &quot;t204&quot; ...
##   ..$ : chr [1:456] &quot;t862&quot; &quot;t409&quot; &quot;t260&quot; &quot;t204&quot; ...</code></pre>
</div>
<div id="mpd-anchored-to-the-root" class="section level4">
<h4><span class="header-section-number">5.1.3.4</span> MPD anchored to the root</h4>
<p>There is an extention to mean pairwise distance calculations from <span class="citation">Helmus et al. (<a href="#ref-Helmus2010">2010</a>)</span> called <span class="math inline">\(PSV\)</span>, <span class="math inline">\(PSR\)</span>, and <span class="math inline">\(PSE\)</span>, phylogenetic species variability, phylogenetic species richness, and phylogenetic species evenness. These measures take the basic pairwise distance calculations and anchor them to the root of the tree so distances have a common denominator. This extention is implemented by using the same equations, just with a constrained set of <span class="math inline">\(d_{ij}\)</span> conditions. Specifically,</p>
<p><span class="math display">\[PSV = MPD = \dfrac{\sum_{ij} d_{ij}}{S(S-1)}\]</span> <span class="math display">\[PSR = \sum_{i} {(\dfrac{1}{S-1} \sum_{j} {d_{ij}})}\]</span> <span class="math display">\[PSE = \dfrac{S}{S-1} \sum_{ij} d_{ij}p_{i}p_{j}\]</span></p>
<p>with these specific values of <span class="math inline">\(d_{ij}\)</span></p>
<p><span class="math display">\[
d_{ji}=0.5*(c_{ii} + c_{jj} - c_{ij}) \\ 
\ \\
or \\
\ \\
d_{ij} = 1 - c_{ij} / (\sqrt{c_{ii}c_{jj}}) 
\]</span> and <span class="math display">\[
c_{ii} = the \ sum \ of \ branch \ lengths \ from \ tip \ i  \ to \ the \ root \ of \ the \ phylogenetic \ tree. \\
\ \\
c_{ij} = the \ sum \  of \ branch \ lengths \ from \ first \ common \ ancestor \ for \ i \ and \ j \ to \ the \ root.
\]</span></p>
</div>
<div id="average-distance-between-two-randomly-chosen-species" class="section level4">
<h4><span class="header-section-number">5.1.3.5</span> Average distance between two randomly chosen species</h4>
<p><span class="math inline">\(J\)</span>, Intensive quadratic entropy, which is the average distance between two randomly chosen species <span class="citation">(Izsak and Papp <a href="#ref-Izsak2000">2000</a>)</span> <span class="math display">\[J = \dfrac{\sum_{ij}d_{ij}}{S^2} \]</span></p>
</div>
<div id="simpsons-diversity-index-for-pairwise-distance" class="section level4">
<h4><span class="header-section-number">5.1.3.6</span> Simpson’s diversity index for pairwise distance</h4>
<p>There has been a long effort to pen a phylogenetic analogy to a Simpson’s diversity index. <span class="citation">(Rao <a href="#ref-Rao1982">1982</a>; Clarke and Warwick <a href="#ref-Clarke1998">1998</a>; S. Ã. Pavoine, Ollier, and Pontier <a href="#ref-Pavoine2005">2005</a>; Hardy and Senterre <a href="#ref-Hardy2007">2007</a>; Webb et al. <a href="#ref-Webb2002">2002</a>; Webb, Ackerly, and Kembel <a href="#ref-Webb2008">2008</a>; Kembel et al. <a href="#ref-Kembel2010">2010</a>)</span>. The conclusion seems to be that this measure is equivilent to scaling <span class="math inline">\(MPD\)</span> by abundance <span class="math inline">\(p_{i}\)</span> and <span class="math inline">\(p_{j}\)</span> to get <span class="math inline">\(MPD_{Ab}\)</span>. This is also a special case of Rao’s Quadratic Entropy, <span class="math inline">\(Roe&#39;s QE\)</span>. Note: not using abundance measures yet for D-place data.</p>
<p><span class="math display">\[MPD_{Ab} = \sum_{i} \sum_{j} d_{ij} p_{i} p_{j}\]</span></p>
</div>
<div id="interspecific-comparisons-of-pairwise-distances" class="section level4">
<h4><span class="header-section-number">5.1.3.7</span> Interspecific comparisons of pairwise distances</h4>
<p>The interspecific variant (rather than the intraspecific default described above) defines the expected phylogenetic distance between two indivdiuals randomely drawn conditionally on the fact that they indivdulas from different species. <span class="math display">\[InterMPD_{Ab} = \dfrac{\sum_{i} \sum_{j \ne i} d_{ij} p_{i} p_{j} }{\sum_{i} \sum_{j \ne i} d_{ij} p_{i} p_{j}}  \]</span></p>
</div>
<div id="variance-in-pairwise-distances-vpd" class="section level4">
<h4><span class="header-section-number">5.1.3.8</span> Variance in pairwise distances (<span class="math inline">\(VPD\)</span>)</h4>
<p>Variance in pairwise distances, <span class="math inline">\(VPD\)</span> (a.k.a. <span class="math inline">\(VarTD\)</span> and <span class="math inline">\(\Lambda^+\)</span>), is a regularity indices. <span class="citation">Clarke and Warwick (<a href="#ref-Clarke2001">2001</a>)</span> Variance is relative to tips, <span class="math inline">\(S\)</span>, not to total branches (<span class="math inline">\(B\)</span> from above). These are the residuals, they compare each individual pairwise connection to the overall mean.</p>
<p><span class="math display">\[VPD = \dfrac{1}{S(S-1)} (\sum_{i} \sum_{j \ne i} {(d_{ij} - MPD)^2})\]</span></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#need to adjust to equation above!</span>

<span class="co">#Pairwise distance/all distances -- Variance of pairwise distances</span>

<span class="co"># Anchor test = VPD (variation of pairwise distance)  </span>

variance_pairwise_distance &lt;-<span class="st"> </span><span class="kw">var</span>(<span class="kw">as.vector</span>(Pairwise_dist))</code></pre></div>
<p>Variants of <span class="math inline">\(VPD\)</span> are <span class="math inline">\(VPD_{ab}\)</span> and <span class="math inline">\(InterPVD_{Ab}\)</span>, where variance is scaled by abundance or compared in and out of species. These are also regularity indices.</p>
<p><span class="math display">\[
VPD_{Ab}= 
(\sum_{i} \sum_{j} n_{i} n_{j}) *
\dfrac{\sum_{i} \sum_{j} n_{i} n_{j} (d_{ij} - MPD_{Ab})^2}
{(\sum_{i} \sum_{j} n_{i} n_{j})^2 - \sum_{i} \sum_{j} (n_{i} n_{j})^2}
\\
or
\\
InterVPD_{Ab} = 
(\sum_{i} \sum_{j \ne i} n_{i} n_{j}) *
\dfrac{\sum_{i} \sum_{j \ne i} n_{i} n_{j} (d_{ij} - InterMPD_{Ab})^2}
{(\sum_{i} \sum_{j \ne i} n_{i} n_{j})^2 - \sum_{i} \sum_{j \ne i} (n_{i} n_{j})^2}
\]</span></p>
</div>
</div>
<div id="nearest-phylogenetic-neighbor" class="section level3">
<h3><span class="header-section-number">5.1.4</span> Nearest phylogenetic neighbor</h3>
<div id="divergence-indices" class="section level4">
<h4><span class="header-section-number">5.1.4.1</span> Divergence indices</h4>
<p>Divergence indices using nearest distance: <span class="math inline">\(MNTD\)</span> and <span class="math inline">\(MNTD_{Ab}\)</span>, Mean nearest taxon distance and Abundance-weighted MNTD <span class="citation">(Webb et al. <a href="#ref-Webb2002">2002</a>; Webb, Ackerly, and Kembel <a href="#ref-Webb2008">2008</a>; Kembel et al. <a href="#ref-Kembel2010">2010</a>)</span>.</p>
<p><span class="math inline">\(MNTD\)</span>, mean nearest taxon distance, is the mean shortest distance from a species to all other in the assemblage <span class="citation">(Webb et al. <a href="#ref-Webb2002">2002</a>; Webb, Ackerly, and Kembel <a href="#ref-Webb2008">2008</a>; Kembel et al. <a href="#ref-Kembel2010">2010</a>)</span>.<br />
<span class="math display">\[
MNTD = 
\dfrac{1}{S}
\sum_{i}
d_{i_{min}}
\]</span></p>
<p><span class="math inline">\(MNTD_{AB}\)</span>, abundance adjusted mean nearest taxon distance. Adjusted by species proportions (i.e. species’ relative abundances) <span class="citation">(Webb et al. <a href="#ref-Webb2002">2002</a>; Webb, Ackerly, and Kembel <a href="#ref-Webb2008">2008</a>; Kembel et al. <a href="#ref-Kembel2010">2010</a>)</span><br />
<span class="math display">\[
MNTD_{Ab} = 
\sum_{i=1}^{S}
[d_{i_{min}} * p_{i}]
\]</span></p>
</div>
<div id="regularity-indices" class="section level4">
<h4><span class="header-section-number">5.1.4.2</span> Regularity indices</h4>
<p>Regularity indices using nearest distances: <span class="math inline">\(VNTD\)</span>, <span class="math inline">\(VNTD_{Ab}\)</span>, <span class="math inline">\(PE_{ev}\)</span>.</p>
<p><span class="math inline">\(VNTD\)</span>, Variance in nearest taxon distances, is the variance in nearest pairwise distance <span class="citation">(Tucker et al. <a href="#ref-Tucker2016">2016</a>)</span>. <span class="math display">\[
VNTD = \dfrac{1}{S}
\sum_{i-1}^{S}
[(d_{i_{min}} - MNTD)^2]
\]</span> <span class="math inline">\(VNTD_{Ab}\)</span>, Abundance weighted variance in nearest taxon distances, is scales by abundance in the same way as descried above <span class="citation">(Tucker et al. <a href="#ref-Tucker2016">2016</a>)</span>. <span class="math display">\[
VNTD_{Ab} = \dfrac
{(\sum_{i} n_{i}) \sum_{i} n_{i} (d_{i_{min}} - MNTD_{Ab})^2}
{(\sum_{i} n_{i})^2 - \sum_{i} n_{i} ^2}
\]</span></p>
</div>
<div id="phylogenetic-version-of-the-funtional-fe_ve-index" class="section level4">
<h4><span class="header-section-number">5.1.4.3</span> Phylogenetic version of the funtional <span class="math inline">\(FE_{ve}\)</span> index</h4>
<p><span class="math inline">\(PE_{ve}\)</span>, phylogenetic evenness is a phylogenetic version of the funtional <span class="math inline">\(FE_{ve}\)</span> index. First a minimum spanning tree (<span class="math inline">\(MST\)</span>) is computed using the cophenetic distance obtained from the phylogenetic tree. The <span class="math inline">\(MST\)</span> contains <span class="math inline">\(S-1\)</span> Branches connection the <span class="math inline">\(S\)</span> species. We denote <span class="math inline">\(l\)</span> a branch on the <span class="math inline">\(MST\)</span>, <span class="math inline">\(dist(i,j)\)</span> is the length the branch <span class="math inline">\(l\)</span> that connects species <span class="math inline">\(i\)</span> and <span class="math inline">\(j\)</span>. <span class="math inline">\(n_{i}\)</span> is, as defined above, the abundance of species <span class="math inline">\(i\)</span> in the asseblage <span class="citation">(Villeger, Mason, and Mouillot <a href="#ref-Villeger2008">2008</a>; Dehling et al. <a href="#ref-Dehling2014">2014</a>)</span>.</p>
<p><span class="math display">\[
Weighted \ evenness: \\
EW_{i} = \dfrac{dist(i,j)}
{(n_{i} + n_{j})/(\sum_{k=1}^{S}n_{k})} \\
\ \\
Partial \ weighted \ evenness: \\ 
PEW_{l} = \dfrac
{EW_{l}}
{\sum_{l=1}^{S-1} EW_{l}} \\
\ \\
Phylogenetic \ evenness: \\
PE_{ve} = \dfrac
{\sum_{l=1}^{S-1} min(PEW_{l}, \dfrac{1}{S-1}) - (\dfrac{1}{S-1})}
{1- (\dfrac{1}{S-1})}
\]</span></p>
</div>
</div>
<div id="phylogenetic-isolation" class="section level3">
<h3><span class="header-section-number">5.1.5</span> Phylogenetic isolation</h3>
<p>A phylogenetic isolation index represents the relative isolation of a given species within a phylogenetic tree. Several indices have been proposed so far but we focus here on the evolutionary distinctiveness index called ‘Fair Proportion’ as proposed by <span class="citation">Redding (<a href="#ref-Redding2003">2003</a>)</span> and <span class="citation">Isaac et al. (<a href="#ref-Isaac2007">2007</a>)</span>.</p>
<div id="evolutionary-distinctiveness-richness-indices" class="section level4">
<h4><span class="header-section-number">5.1.5.1</span> Evolutionary distinctiveness (richness indices)</h4>
<p><span class="math inline">\(ED\)</span>, evolutionary distinctiveness is a richness indices. NOTE: not equal to Faith’s PD because the <span class="math inline">\(ED_{i}\)</span> are computed from the regional pool of species and sumed across a given assemblage (i.e. a subset of the regional species pool) <span class="citation">(Tucker et al. <a href="#ref-Tucker2016">2016</a>; Safi et al. <a href="#ref-Safi2013a">2013</a>; Redding <a href="#ref-Redding2003">2003</a>; Isaac et al. <a href="#ref-Isaac2007">2007</a>)</span>.</p>
<p><span class="math display">\[
  ED = \sum_{i}ED_{i} \\
  \ \\ 
  where \ ED_{i} = \sum_{b \in B_{t_{i}}} \dfrac{L_{b}}{S_{b}} 
\]</span></p>
<p><span class="math inline">\(AED\)</span>, Abundance-weighted <span class="math inline">\(ED\)</span> <span class="citation">(Tucker et al. <a href="#ref-Tucker2016">2016</a>; Cadotte et al. <a href="#ref-Cadotte2010">2010</a>)</span>. <span class="math display">\[
\sum_{i} AED_{i} \\
\ \\
where \ AED_{i} = \sum_{b \in B_{t_{i}}} \dfrac{L_{b}}{A_{b} }* p_{i}
\]</span></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Bruno&#39;s function for ED. Provided in library(FARM)</span>

evol.distinct2 &lt;-<span class="st"> </span><span class="cf">function</span> (tree, <span class="dt">type =</span> <span class="kw">c</span>(<span class="st">&quot;equal.splits&quot;</span>, <span class="st">&quot;fair.proportion&quot;</span>), 
    <span class="dt">scale =</span> <span class="ot">FALSE</span>, <span class="dt">use.branch.lengths =</span> <span class="ot">TRUE</span>) 
{
    type &lt;-<span class="st"> </span><span class="kw">match.arg</span>(type)
    <span class="cf">if</span> (<span class="kw">is.rooted</span>(tree) <span class="op">==</span><span class="st"> </span><span class="ot">FALSE</span>) 
        <span class="kw">warning</span>(<span class="st">&quot;A rooted phylogeny is required for meaningful output of this function&quot;</span>, 
            <span class="dt">call. =</span> <span class="ot">FALSE</span>)
    <span class="cf">if</span> (scale <span class="op">==</span><span class="st"> </span><span class="ot">TRUE</span>) {
        <span class="cf">if</span> (<span class="kw">is.ultrametric</span>(tree) <span class="op">==</span><span class="st"> </span><span class="ot">TRUE</span>) 
            tree<span class="op">$</span>edge.length &lt;-<span class="st"> </span>tree<span class="op">$</span>edge.length<span class="op">/</span>(<span class="kw">as.numeric</span>(<span class="kw">branching.times</span>(tree)[<span class="dv">1</span>]))
        <span class="cf">else</span> tree<span class="op">$</span>edge.length &lt;-<span class="st"> </span>tree<span class="op">$</span>edge.length<span class="op">/</span><span class="kw">sum</span>(tree<span class="op">$</span>edge.length)
    }
    <span class="cf">if</span> (use.branch.lengths <span class="op">==</span><span class="st"> </span><span class="ot">FALSE</span>) 
        tree<span class="op">$</span>edge.length &lt;-<span class="st"> </span><span class="kw">rep</span>(<span class="dv">1</span>, <span class="kw">length</span>(tree<span class="op">$</span>edge.length))
    <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(tree<span class="op">$</span>tip.label)) {
        spp &lt;-<span class="st"> </span>tree<span class="op">$</span>tip.label[i]
        nodes &lt;-<span class="st"> </span><span class="kw">.get.nodes</span>(tree, spp)
        nodes &lt;-<span class="st"> </span>nodes[<span class="dv">1</span><span class="op">:</span>(<span class="kw">length</span>(nodes) <span class="op">-</span><span class="st"> </span><span class="dv">1</span>)]
        internal.brlen &lt;-<span class="st"> </span>tree<span class="op">$</span>edge.length[<span class="kw">which</span>(tree<span class="op">$</span>edge[, 
            <span class="dv">2</span>] <span class="op">%in%</span><span class="st"> </span>nodes)]
        <span class="cf">if</span> (<span class="kw">length</span>(internal.brlen) <span class="op">!=</span><span class="st"> </span><span class="dv">0</span>) {
            internal.brlen &lt;-<span class="st"> </span>internal.brlen <span class="op">*</span><span class="st"> </span><span class="cf">switch</span>(type, <span class="dt">equal.splits =</span> <span class="kw">sort</span>(<span class="kw">rep</span>(<span class="fl">0.5</span>, 
                <span class="kw">length</span>(internal.brlen))<span class="op">^</span><span class="kw">c</span>(<span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(internal.brlen))), 
                <span class="dt">fair.proportion =</span> {
                  <span class="cf">for</span> (j <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(nodes)) {
                    sons &lt;-<span class="st"> </span><span class="kw">.node.desc</span>(tree, nodes[j])
                    n.descendents &lt;-<span class="st"> </span><span class="kw">length</span>(sons<span class="op">$</span>tips)
                    <span class="cf">if</span> (j <span class="op">==</span><span class="st"> </span><span class="dv">1</span>) portion &lt;-<span class="st"> </span>n.descendents <span class="cf">else</span> portion &lt;-<span class="st"> </span><span class="kw">c</span>(n.descendents, 
                      portion)
                  }
                  <span class="dv">1</span><span class="op">/</span>portion
                })
        }
        ED &lt;-<span class="st"> </span><span class="kw">sum</span>(internal.brlen, tree<span class="op">$</span>edge.length[<span class="kw">which.edge</span>(tree, 
            spp)])
        <span class="cf">if</span> (i <span class="op">==</span><span class="st"> </span><span class="dv">1</span>) 
            w &lt;-<span class="st"> </span>ED
        <span class="cf">else</span> w &lt;-<span class="st"> </span><span class="kw">c</span>(w, ED)
    }
    <span class="kw">return</span>(w)
}</code></pre></div>
<p>Evolutionary distinctiveness is our basic measure of phylogenetic isolation. #This should likely be ‘fair proportions’ instead of ‘equal.splits’.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Calculate ED</span>
<span class="co"># Using equal.splits method, faster computation</span>
<span class="co"># Evolutionary_distinctiveness_i &lt;- evol.distinct2(this_tree, type = &quot;equal.splits&quot;)  </span>

<span class="co"># ED - Summed evolutionary distinctiveness</span>
<span class="co"># Evolutionary_distinctiveness_sum &lt;- sum(Evolutionary_distinctiveness_i)</span></code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#Evolutionary_distinctiveness_sum</span></code></pre></div>
<p>We can run some standard summary statistics (mean and variance) on this ED measure. var(Ed) shows up close to VPD on the PCAs in the intro <span class="citation">(Tucker et al. <a href="#ref-Tucker2016">2016</a>)</span>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># mean(ED)</span>
<span class="co"># mean_Phylogenetic_isolation &lt;- mean(Evolutionary_distinctiveness_i)</span>

<span class="co"># var(ED)</span>
<span class="co">#variance_Phylogenetic_isolation &lt;- var(Evolutionary_distinctiveness_i)</span></code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#mean_Phylogenetic_isolation</span>
<span class="co">#variance_Phylogenetic_isolation</span></code></pre></div>
</div>
<div id="mean-evolutionary-distinctiveness-divergence-indices" class="section level4">
<h4><span class="header-section-number">5.1.5.2</span> Mean evolutionary distinctiveness (divergence indices)</h4>
<p>The divergence indices version for <span class="math inline">\(ED\)</span> is mean evolutionary distinctiveness, <span class="math inline">\(MED\)</span>. The mean of evolutionary distinctiveness <span class="citation">(Redding <a href="#ref-Redding2003">2003</a>; Isaac et al. <a href="#ref-Isaac2007">2007</a>)</span>. <span class="math display">\[
MED = \dfrac
{\sum_{i} ED_{i}}
{S} \\
\ \\
with \\
\ \\
ED_{i} = \sum_{b \in B_{t_{i}}} \dfrac{L_{b}}{S_{b}}
\]</span> #### Entropy measure of evolutonary distinctiveness (regularity indices) The regularity indices for <span class="math inline">\(ED\)</span>/phylogenetic isolation are <span class="math inline">\(H_{ED}\)</span>, <span class="math inline">\(E_{ED}\)</span>, <span class="math inline">\(var(ED)\)</span>, <span class="math inline">\(H_{AED}\)</span></p>
<p><span class="math inline">\(H_{ED}\)</span>, Entropy measure of evolutionary distinctiveness, is the shannon index applied to evolutionary distinctiveness values <span class="citation">(Cadotte et al. <a href="#ref-Cadotte2010">2010</a>)</span>. <span class="math display">\[
H_{ED} = -\sum_{i=1}^{S}
((\dfrac{ED_{i}}{\sum_{i=1}^{S} ED_{i}})
* \ln (\dfrac{ED_{i}}{\sum_{i=1}^{S} ED_{i}}))
\]</span></p>
<p><span class="math inline">\(E_{ED}\)</span>, Equitability of evolutionary distinctiveness, is <span class="math inline">\(H_{ED}\)</span> controlled for species richness <span class="citation">(Cadotte et al. <a href="#ref-Cadotte2010">2010</a>)</span>.</p>
<p><span class="math display">\[
E_{ED} = \dfrac{H_{ED}}{\ln(S)}
\]</span> <span class="math inline">\(var(ED)\)</span>, Variance in evolutinoary distinctiveness, is the variance of species evolutionary distinctiveness <span class="citation">(Tucker et al. <a href="#ref-Tucker2016">2016</a>)</span>.</p>
<p><span class="math display">\[
var(ED) = \dfrac{1}{S-1} *
\sum_{i=1}^{S}
(ED_{i}-\dfrac{\sum_{i=1}^{S} ED_{i}}{S})^2
\]</span> <span class="math inline">\(H_{ED_{Ab}}\)</span>, Abundance-weighted version of <span class="math inline">\(H_{ED}\)</span> <span class="citation">(Cadotte et al. <a href="#ref-Cadotte2010">2010</a>)</span>.</p>
<p><span class="math display">\[
H_{ED_{Ab}} = -\sum_{i=1}^{S}
(\dfrac{n_{i}AED_{i}}{\sum_{i=1}^{S} n_{i}AED_{i}} *
\ln(\dfrac{n_{i}AED_{i}}{\sum_{i=1}^{S} n_{i}AED_{i}}))
\]</span></p>
</div>
</div>
</div>
<div id="beta-diversity" class="section level2">
<h2><span class="header-section-number">5.2</span> Beta diversity</h2>
<p>We currently are not using any beta diversity metrics but there are many to choose from if we decide to add them later.</p>
</div>
<div id="tree-topology" class="section level2">
<h2><span class="header-section-number">5.3</span> Tree topology</h2>
<p>Tree topology is a measure of the shape of the overall tree. The tree can be lopsided side-to-side or front-to-back.</p>
<p>Our most trusted index for the tippy vs trunky of a tree is the gamma index, <span class="math inline">\(\gamma\)</span>.The index characterizes the distribution of branching events within the tree. Trees with <span class="math inline">\(\gamma &lt; 0\)</span> have relatively longer branches towards the tips of the phylogeny (tippy trees), whereas trees with <span class="math inline">\(\gamma &gt; 0\)</span> have relatively longer inter-nodal distances towards the root of the phylogeny (stemmy trees). tk represents an ‘evolutionary period’ (limits are given by two speciation events) or equivalently an internode distance <span class="citation">(Pybus and Harvey <a href="#ref-Pybus2000">2000</a>)</span>.</p>
<p><span class="math display">\[
  \gamma = \dfrac
  {(\dfrac{1}{S-2}* \sum_{i=2}^{S-1} (\sum_{k=2}^{i} Kt_{k}))- \dfrac{1}{2} * \sum_{j=2}^{S} jt_{j}}
  {(\sum_{j=2}^{S} jt_{j}) * \sqrt{\dfrac{1}{12*(S-2)}}}
\]</span></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># ltt function from library(phytools)</span>
ltt &lt;-<span class="st"> </span><span class="cf">function</span> (tree, <span class="dt">plot =</span> <span class="ot">TRUE</span>, <span class="dt">drop.extinct =</span> <span class="ot">FALSE</span>, <span class="dt">log.lineages =</span> <span class="ot">TRUE</span>, 
    <span class="dt">gamma =</span> <span class="ot">TRUE</span>, ...) 
{
    tol &lt;-<span class="st"> </span><span class="fl">1e-06</span>
    <span class="cf">if</span> (<span class="op">!</span><span class="kw">inherits</span>(tree, <span class="st">&quot;phylo&quot;</span>) <span class="op">&amp;&amp;</span><span class="st"> </span><span class="op">!</span><span class="kw">inherits</span>(tree, <span class="st">&quot;multiPhylo&quot;</span>)) 
        <span class="kw">stop</span>(<span class="st">&quot;tree must be object of class </span><span class="ch">\&quot;</span><span class="st">phylo</span><span class="ch">\&quot;</span><span class="st"> or </span><span class="ch">\&quot;</span><span class="st">multiPhylo</span><span class="ch">\&quot;</span><span class="st">.&quot;</span>)
    <span class="cf">if</span> (<span class="kw">inherits</span>(tree, <span class="st">&quot;multiPhylo&quot;</span>)) {
        obj &lt;-<span class="st"> </span><span class="kw">lapply</span>(tree, ltt, <span class="dt">plot =</span> <span class="ot">FALSE</span>, <span class="dt">drop.extinct =</span> drop.extinct, 
            <span class="dt">log.lineages =</span> log.lineages, <span class="dt">gamma =</span> gamma)
        <span class="kw">class</span>(obj) &lt;-<span class="st"> &quot;multiLtt&quot;</span>
    }
    <span class="cf">else</span> {
        tree &lt;-<span class="st"> </span><span class="kw">reorder.phylo</span>(tree, <span class="dt">order =</span> <span class="st">&quot;cladewise&quot;</span>)
        <span class="cf">if</span> (<span class="op">!</span><span class="kw">is.null</span>(tree<span class="op">$</span>node.label)) {
            node.names &lt;-<span class="st"> </span><span class="kw">setNames</span>(tree<span class="op">$</span>node.label, <span class="dv">1</span><span class="op">:</span>tree<span class="op">$</span>Nnode <span class="op">+</span><span class="st"> </span>
<span class="st">                </span><span class="kw">Ntip</span>(tree))
            tree<span class="op">$</span>node.label &lt;-<span class="st"> </span><span class="ot">NULL</span>
        }
        <span class="cf">else</span> node.names &lt;-<span class="st"> </span><span class="ot">NULL</span>
        <span class="cf">if</span> (<span class="kw">is.ultrametric</span>(tree)) {
            h &lt;-<span class="st"> </span><span class="kw">max</span>(<span class="kw">nodeHeights</span>(tree))
            time &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dv">0</span>, h <span class="op">-</span><span class="st"> </span><span class="kw">sort</span>(<span class="kw">branching.times</span>(tree), <span class="dt">decreasing =</span> <span class="ot">TRUE</span>), 
                h)
            nodes &lt;-<span class="st"> </span><span class="kw">as.numeric</span>(<span class="kw">names</span>(time)[<span class="dv">2</span><span class="op">:</span>(<span class="kw">length</span>(time) <span class="op">-</span><span class="st"> </span>
<span class="st">                </span><span class="dv">1</span>)])
            ltt &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="kw">cumsum</span>(<span class="kw">c</span>(<span class="dv">1</span>, <span class="kw">sapply</span>(nodes, <span class="cf">function</span>(x, y) <span class="kw">sum</span>(y <span class="op">==</span><span class="st"> </span>
<span class="st">                </span>x) <span class="op">-</span><span class="st"> </span><span class="dv">1</span>, <span class="dt">y =</span> tree<span class="op">$</span>edge[, <span class="dv">1</span>]))), <span class="kw">length</span>(tree<span class="op">$</span>tip.label))
            <span class="kw">names</span>(ltt) &lt;-<span class="st"> </span><span class="kw">names</span>(time)
        }
        <span class="cf">else</span> {
            drop.extinct.tips &lt;-<span class="st"> </span><span class="cf">function</span>(phy) {
                temp &lt;-<span class="st"> </span><span class="kw">diag</span>(<span class="kw">vcv</span>(phy))
                <span class="cf">if</span> (<span class="kw">length</span>(temp[temp <span class="op">&lt;</span><span class="st"> </span>(<span class="kw">max</span>(temp) <span class="op">-</span><span class="st"> </span>tol)]) <span class="op">&gt;</span><span class="st"> </span>
<span class="st">                  </span><span class="dv">0</span>) 
                  pruned.phy &lt;-<span class="st"> </span><span class="kw">drop.tip</span>(phy, <span class="kw">names</span>(temp[temp <span class="op">&lt;</span><span class="st"> </span>
<span class="st">                    </span>(<span class="kw">max</span>(temp) <span class="op">-</span><span class="st"> </span>tol)]))
                <span class="cf">else</span> pruned.phy &lt;-<span class="st"> </span>phy
                <span class="kw">return</span>(pruned.phy)
            }
            <span class="cf">if</span> (drop.extinct <span class="op">==</span><span class="st"> </span><span class="ot">TRUE</span>) 
                tree &lt;-<span class="st"> </span><span class="kw">drop.extinct.tips</span>(tree)
            root &lt;-<span class="st"> </span><span class="kw">length</span>(tree<span class="op">$</span>tip) <span class="op">+</span><span class="st"> </span><span class="dv">1</span>
            node.height &lt;-<span class="st"> </span><span class="kw">matrix</span>(<span class="ot">NA</span>, <span class="kw">nrow</span>(tree<span class="op">$</span>edge), <span class="dv">2</span>)
            <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">nrow</span>(tree<span class="op">$</span>edge)) {
                <span class="cf">if</span> (tree<span class="op">$</span>edge[i, <span class="dv">1</span>] <span class="op">==</span><span class="st"> </span>root) {
                  node.height[i, <span class="dv">1</span>] &lt;-<span class="st"> </span><span class="dv">0</span>
                  node.height[i, <span class="dv">2</span>] &lt;-<span class="st"> </span>tree<span class="op">$</span>edge.length[i]
                }
                <span class="cf">else</span> {
                  node.height[i, <span class="dv">1</span>] &lt;-<span class="st"> </span>node.height[<span class="kw">match</span>(tree<span class="op">$</span>edge[i, 
                    <span class="dv">1</span>], tree<span class="op">$</span>edge[, <span class="dv">2</span>]), <span class="dv">2</span>]
                  node.height[i, <span class="dv">2</span>] &lt;-<span class="st"> </span>node.height[i, <span class="dv">1</span>] <span class="op">+</span><span class="st"> </span>tree<span class="op">$</span>edge.length[i]
                }
            }
            ltt &lt;-<span class="st"> </span><span class="kw">vector</span>()
            tree.length &lt;-<span class="st"> </span><span class="kw">max</span>(node.height)
            n.extinct &lt;-<span class="st"> </span><span class="kw">sum</span>(node.height[tree<span class="op">$</span>edge[, <span class="dv">2</span>] <span class="op">&lt;=</span><span class="st"> </span><span class="kw">length</span>(tree<span class="op">$</span>tip), 
                <span class="dv">2</span>] <span class="op">&lt;</span><span class="st"> </span>(tree.length <span class="op">-</span><span class="st"> </span>tol))
            node.height[tree<span class="op">$</span>edge[, <span class="dv">2</span>] <span class="op">&lt;=</span><span class="st"> </span><span class="kw">length</span>(tree<span class="op">$</span>tip), <span class="dv">2</span>] &lt;-<span class="st"> </span>node.height[tree<span class="op">$</span>edge[, 
                <span class="dv">2</span>] <span class="op">&lt;=</span><span class="st"> </span><span class="kw">length</span>(tree<span class="op">$</span>tip), <span class="dv">2</span>] <span class="op">+</span><span class="st"> </span><span class="fl">1.1</span> <span class="op">*</span><span class="st"> </span>tol
            time &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dv">0</span>, node.height[, <span class="dv">2</span>])
            <span class="kw">names</span>(time) &lt;-<span class="st"> </span><span class="kw">as.character</span>(<span class="kw">c</span>(root, tree<span class="op">$</span>edge[, <span class="dv">2</span>]))
            temp &lt;-<span class="st"> </span><span class="kw">vector</span>()
            time &lt;-<span class="st"> </span>time[<span class="kw">order</span>(time)]
            time &lt;-<span class="st"> </span>time[<span class="dv">1</span><span class="op">:</span>(tree<span class="op">$</span>Nnode <span class="op">+</span><span class="st"> </span>n.extinct <span class="op">+</span><span class="st"> </span><span class="dv">1</span>)]
            <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span>(<span class="kw">length</span>(time) <span class="op">-</span><span class="st"> </span><span class="dv">1</span>)) {
                ltt[i] &lt;-<span class="st"> </span><span class="dv">0</span>
                <span class="cf">for</span> (j <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">nrow</span>(node.height)) ltt[i] &lt;-<span class="st"> </span>ltt[i] <span class="op">+</span><span class="st"> </span>
<span class="st">                  </span>(time[i] <span class="op">&gt;=</span><span class="st"> </span>(node.height[j, <span class="dv">1</span>] <span class="op">-</span><span class="st"> </span>tol) <span class="op">&amp;&amp;</span><span class="st"> </span>time[i] <span class="op">&lt;=</span><span class="st"> </span>
<span class="st">                    </span>(node.height[j, <span class="dv">2</span>] <span class="op">-</span><span class="st"> </span>tol))
            }
            ltt[i <span class="op">+</span><span class="st"> </span><span class="dv">1</span>] &lt;-<span class="st"> </span><span class="dv">0</span>
            <span class="cf">for</span> (j <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">nrow</span>(node.height)) ltt[i <span class="op">+</span><span class="st"> </span><span class="dv">1</span>] &lt;-<span class="st"> </span>ltt[i <span class="op">+</span><span class="st"> </span>
<span class="st">                </span><span class="dv">1</span>] <span class="op">+</span><span class="st"> </span>(time[i <span class="op">+</span><span class="st"> </span><span class="dv">1</span>] <span class="op">&lt;=</span><span class="st"> </span>(node.height[j, <span class="dv">2</span>] <span class="op">+</span><span class="st"> </span>tol))
            <span class="kw">names</span>(ltt) &lt;-<span class="st"> </span><span class="kw">names</span>(time)
            ltt &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dv">1</span>, ltt)
            time &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dv">0</span>, time)
            time[<span class="kw">length</span>(time)] &lt;-<span class="st"> </span>time[<span class="kw">length</span>(time)] <span class="op">-</span><span class="st"> </span><span class="fl">1.1</span> <span class="op">*</span><span class="st"> </span>
<span class="st">                </span>tol
        }
        <span class="cf">if</span> (<span class="op">!</span><span class="kw">is.null</span>(node.names)) {
            nn &lt;-<span class="st"> </span><span class="kw">sapply</span>(<span class="kw">names</span>(time), <span class="cf">function</span>(x, y) <span class="cf">if</span> (<span class="kw">any</span>(<span class="kw">names</span>(y) <span class="op">==</span><span class="st"> </span>
<span class="st">                </span>x)) 
                y[<span class="kw">which</span>(<span class="kw">names</span>(y) <span class="op">==</span><span class="st"> </span>x)]
            <span class="cf">else</span> <span class="st">&quot;&quot;</span>, <span class="dt">y =</span> node.names)
            <span class="kw">names</span>(ltt) &lt;-<span class="st"> </span><span class="kw">names</span>(time) &lt;-<span class="st"> </span>nn
        }
        <span class="cf">if</span> (gamma <span class="op">==</span><span class="st"> </span><span class="ot">FALSE</span>) {
            obj &lt;-<span class="st"> </span><span class="kw">list</span>(<span class="dt">ltt =</span> ltt, <span class="dt">times =</span> time, <span class="dt">tree =</span> tree)
            <span class="kw">class</span>(obj) &lt;-<span class="st"> &quot;ltt&quot;</span>
        }
        <span class="cf">else</span> {
            gam &lt;-<span class="st"> </span><span class="kw">gammatest</span>(<span class="kw">list</span>(<span class="dt">ltt =</span> ltt, <span class="dt">times =</span> time))
            obj &lt;-<span class="st"> </span><span class="kw">list</span>(<span class="dt">ltt =</span> ltt, <span class="dt">times =</span> time, <span class="dt">gamma =</span> gam<span class="op">$</span>gamma, 
                <span class="dt">p =</span> gam<span class="op">$</span>p, <span class="dt">tree =</span> tree)
            <span class="kw">class</span>(obj) &lt;-<span class="st"> &quot;ltt&quot;</span>
        }
    }
    <span class="cf">if</span> (plot) 
        <span class="kw">plot</span>(obj, <span class="dt">log.lineages =</span> log.lineages, ...)
    obj
}
<span class="op">&lt;</span>environment<span class="op">:</span><span class="st"> </span>namespace<span class="op">:</span>phytools<span class="op">&gt;</span></code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">ltts &lt;-<span class="st"> </span><span class="kw">ltt</span>(this_tree, <span class="dt">gamma =</span> <span class="ot">TRUE</span>, <span class="dt">plot =</span> <span class="ot">FALSE</span>)
ltts</code></pre></div>
<pre><code>## Object of class &quot;ltt&quot; containing:
## 
## (1) A phylogenetic tree with 456 tips and 455 internal nodes.
## 
## (2) Vectors containing the number of lineages (ltt) and branching times (times) on the tree.
## 
## (3) A value for Pybus &amp; Harvey&#39;s &quot;gamma&quot; statistic of 1.7401, p-value = 0.0818.</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">str</span>(ltts)</code></pre></div>
<pre><code>## List of 5
##  $ ltt  : Named num [1:457] 1 2 3 4 5 6 7 8 9 10 ...
##   ..- attr(*, &quot;names&quot;)= chr [1:457] &quot;&quot; &quot;457&quot; &quot;458&quot; &quot;459&quot; ...
##  $ times: Named num [1:457] 0.00 2.84e-13 1.00 2.00 2.67 ...
##   ..- attr(*, &quot;names&quot;)= chr [1:457] &quot;&quot; &quot;457&quot; &quot;458&quot; &quot;459&quot; ...
##  $ gamma: num 1.74
##  $ p    : num 0.0818
##  $ tree :List of 4
##   ..$ edge       : num [1:910, 1:2] 457 460 463 560 892 892 560 463 466 466 ...
##   ..$ tip.label  : chr [1:456] &quot;t862&quot; &quot;t409&quot; &quot;t260&quot; &quot;t204&quot; ...
##   ..$ edge.length: num [1:910] 2.667 4.433 20.973 10.011 0.916 ...
##   ..$ Nnode      : num 455
##   ..- attr(*, &quot;class&quot;)= chr &quot;phylo&quot;
##   ..- attr(*, &quot;order&quot;)= chr &quot;cladewise&quot;
##  - attr(*, &quot;class&quot;)= chr &quot;ltt&quot;</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">lineages_through_time &lt;-<span class="st"> </span><span class="kw">as.numeric</span>(ltts[[<span class="dv">1</span>]])
time_steps &lt;-<span class="st"> </span><span class="kw">as.numeric</span>(ltts[[<span class="dv">2</span>]])
<span class="co">#extract Gamma index</span>
gamma &lt;-<span class="st"> </span>ltts[[<span class="dv">3</span>]]
gamma_p_value &lt;-<span class="st"> </span>ltts[[<span class="dv">4</span>]]</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">lineages_through_time </code></pre></div>
<pre><code>##   [1]   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17
##  [18]  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34
##  [35]  35  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51
##  [52]  52  53  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68
##  [69]  69  70  71  72  73  74  75  76  77  78  79  80  81  82  83  84  85
##  [86]  86  87  88  89  90  91  92  93  94  95  96  97  98  99 100 101 102
## [103] 103 104 105 106 107 108 109 110 111 112 113 114 115 116 117 118 119
## [120] 120 121 122 123 124 125 126 127 128 129 130 131 132 133 134 135 136
## [137] 137 138 139 140 141 142 143 144 145 146 147 148 149 150 151 152 153
## [154] 154 155 156 157 158 159 160 161 162 163 164 165 166 167 168 169 170
## [171] 171 172 173 174 175 176 177 178 179 180 181 182 183 184 185 186 187
## [188] 188 189 190 191 192 193 194 195 196 197 198 199 200 201 202 203 204
## [205] 205 206 207 208 209 210 211 212 213 214 215 216 217 218 219 220 221
## [222] 222 223 224 225 226 227 228 229 230 231 232 233 234 235 236 237 238
## [239] 239 240 241 242 243 244 245 246 247 248 249 250 251 252 253 254 255
## [256] 256 257 258 259 260 261 262 263 264 265 266 267 268 269 270 271 272
## [273] 273 274 275 276 277 278 279 280 281 282 283 284 285 286 287 288 289
## [290] 290 291 292 293 294 295 296 297 298 299 300 301 302 303 304 305 306
## [307] 307 308 309 310 311 312 313 314 315 316 317 318 319 320 321 322 323
## [324] 324 325 326 327 328 329 330 331 332 333 334 335 336 337 338 339 340
## [341] 341 342 343 344 345 346 347 348 349 350 351 352 353 354 355 356 357
## [358] 358 359 360 361 362 363 364 365 366 367 368 369 370 371 372 373 374
## [375] 375 376 377 378 379 380 381 382 383 384 385 386 387 388 389 390 391
## [392] 392 393 394 395 396 397 398 399 400 401 402 403 404 405 406 407 408
## [409] 409 410 411 412 413 414 415 416 417 418 419 420 421 422 423 424 425
## [426] 426 427 428 429 430 431 432 433 434 435 436 437 438 439 440 441 442
## [443] 443 444 445 446 447 448 449 450 451 452 453 454 455 456 456</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">time_steps </code></pre></div>
<pre><code>##   [1] 0.000000e+00 2.842171e-13 1.000000e+00 2.000000e+00 2.666667e+00
##   [6] 3.333333e+00 5.461538e+00 7.100000e+00 7.450000e+00 9.296296e+00
##  [11] 1.003030e+01 1.024242e+01 1.121429e+01 1.126190e+01 1.200000e+01
##  [16] 1.203509e+01 1.215789e+01 1.217544e+01 1.314815e+01 1.316667e+01
##  [21] 1.335185e+01 1.416667e+01 1.428788e+01 1.436364e+01 1.502222e+01
##  [26] 1.503333e+01 1.512222e+01 1.527778e+01 1.600000e+01 1.630588e+01
##  [31] 1.720588e+01 1.733333e+01 1.734314e+01 1.738235e+01 1.808036e+01
##  [36] 1.833036e+01 1.901613e+01 1.913710e+01 1.915323e+01 1.921774e+01
##  [41] 1.923387e+01 2.006504e+01 2.021138e+01 2.030081e+01 2.038211e+01
##  [46] 2.043902e+01 2.047154e+01 2.100000e+01 2.101361e+01 2.117687e+01
##  [51] 2.130612e+01 2.131973e+01 2.132653e+01 2.204065e+01 2.233333e+01
##  [56] 2.234146e+01 2.236585e+01 2.247967e+01 2.254472e+01 2.314872e+01
##  [61] 2.315897e+01 2.320513e+01 2.406011e+01 2.409290e+01 2.409836e+01
##  [66] 2.424044e+01 2.431694e+01 2.501579e+01 2.503158e+01 2.506842e+01
##  [71] 2.512105e+01 2.514737e+01 2.515263e+01 2.516842e+01 2.517895e+01
##  [76] 2.519474e+01 2.529474e+01 2.534737e+01 2.600985e+01 2.602956e+01
##  [81] 2.604433e+01 2.605419e+01 2.607882e+01 2.623153e+01 2.624138e+01
##  [86] 2.625123e+01 2.625616e+01 2.627586e+01 2.629064e+01 2.630049e+01
##  [91] 2.631034e+01 2.706135e+01 2.711043e+01 2.717791e+01 2.724540e+01
##  [96] 2.725153e+01 2.734356e+01 2.747239e+01 2.749693e+01 2.755828e+01
## [101] 2.757669e+01 2.800385e+01 2.802692e+01 2.804231e+01 2.807308e+01
## [106] 2.813077e+01 2.814615e+01 2.816154e+01 2.816923e+01 2.818462e+01
## [111] 2.820000e+01 2.821923e+01 2.825385e+01 2.830000e+01 2.901115e+01
## [116] 2.901487e+01 2.902230e+01 2.903346e+01 2.905204e+01 2.909294e+01
## [121] 2.912268e+01 2.915985e+01 2.923048e+01 2.938662e+01 3.000683e+01
## [126] 3.001365e+01 3.003413e+01 3.015358e+01 3.018771e+01 3.019795e+01
## [131] 3.021502e+01 3.022184e+01 3.028669e+01 3.032765e+01 3.101894e+01
## [136] 3.104545e+01 3.107955e+01 3.109848e+01 3.115152e+01 3.115909e+01
## [141] 3.117424e+01 3.118939e+01 3.120455e+01 3.120833e+01 3.122348e+01
## [146] 3.125000e+01 3.125758e+01 3.128030e+01 3.129924e+01 3.132197e+01
## [151] 3.138636e+01 3.140909e+01 3.141288e+01 3.141667e+01 3.142803e+01
## [156] 3.201587e+01 3.201905e+01 3.202222e+01 3.202540e+01 3.204127e+01
## [161] 3.205079e+01 3.206984e+01 3.209206e+01 3.210476e+01 3.215873e+01
## [166] 3.216825e+01 3.218095e+01 3.218730e+01 3.223810e+01 3.225714e+01
## [171] 3.228571e+01 3.233651e+01 3.302521e+01 3.303922e+01 3.305042e+01
## [176] 3.305602e+01 3.311204e+01 3.311765e+01 3.312885e+01 3.313445e+01
## [181] 3.314566e+01 3.314846e+01 3.315126e+01 3.315966e+01 3.318487e+01
## [186] 3.319328e+01 3.321289e+01 3.322129e+01 3.322409e+01 3.322689e+01
## [191] 3.322969e+01 3.323249e+01 3.323529e+01 3.323810e+01 3.327171e+01
## [196] 3.328852e+01 3.400290e+01 3.402899e+01 3.405217e+01 3.408406e+01
## [201] 3.409855e+01 3.410725e+01 3.412754e+01 3.413623e+01 3.415652e+01
## [206] 3.417101e+01 3.418261e+01 3.421449e+01 3.422609e+01 3.423478e+01
## [211] 3.425217e+01 3.425797e+01 3.427536e+01 3.427826e+01 3.428986e+01
## [216] 3.430145e+01 3.431304e+01 3.431884e+01 3.432464e+01 3.433043e+01
## [221] 3.435072e+01 3.435942e+01 3.436812e+01 3.500431e+01 3.503664e+01
## [226] 3.504310e+01 3.506897e+01 3.507112e+01 3.508190e+01 3.508836e+01
## [231] 3.509698e+01 3.509914e+01 3.510129e+01 3.510991e+01 3.511853e+01
## [236] 3.600000e+01 3.600651e+01 3.601303e+01 3.602280e+01 3.603257e+01
## [241] 3.603583e+01 3.606189e+01 3.606515e+01 3.606840e+01 3.607166e+01
## [246] 3.607492e+01 3.608143e+01 3.608795e+01 3.609772e+01 3.610098e+01
## [251] 3.610749e+01 3.611401e+01 3.612378e+01 3.614332e+01 3.614984e+01
## [256] 3.615309e+01 3.615635e+01 3.615961e+01 3.617590e+01 3.617915e+01
## [261] 3.618241e+01 3.618567e+01 3.619218e+01 3.619870e+01 3.620195e+01
## [266] 3.620847e+01 3.622150e+01 3.623127e+01 3.623779e+01 3.624104e+01
## [271] 3.624756e+01 3.626384e+01 3.627036e+01 3.629316e+01 3.629967e+01
## [276] 3.630293e+01 3.630945e+01 3.631922e+01 3.632248e+01 3.633225e+01
## [281] 3.634202e+01 3.637134e+01 3.637785e+01 3.640065e+01 3.640391e+01
## [286] 3.640717e+01 3.641368e+01 3.642020e+01 3.642345e+01 3.642671e+01
## [291] 3.642997e+01 3.643322e+01 3.643974e+01 3.644625e+01 3.644951e+01
## [296] 3.645277e+01 3.646254e+01 3.646580e+01 3.647231e+01 3.647557e+01
## [301] 3.648534e+01 3.648860e+01 3.649511e+01 3.650163e+01 3.652117e+01
## [306] 3.653094e+01 3.654397e+01 3.700000e+01 3.700279e+01 3.700559e+01
## [311] 3.700838e+01 3.701117e+01 3.701676e+01 3.702235e+01 3.703073e+01
## [316] 3.703911e+01 3.704749e+01 3.705307e+01 3.705587e+01 3.705866e+01
## [321] 3.706145e+01 3.706704e+01 3.706983e+01 3.707542e+01 3.707821e+01
## [326] 3.708101e+01 3.708380e+01 3.708659e+01 3.710056e+01 3.710335e+01
## [331] 3.710615e+01 3.710894e+01 3.711453e+01 3.711732e+01 3.712011e+01
## [336] 3.712570e+01 3.712849e+01 3.713408e+01 3.713687e+01 3.713966e+01
## [341] 3.714246e+01 3.714804e+01 3.715084e+01 3.715363e+01 3.715642e+01
## [346] 3.715922e+01 3.716201e+01 3.716480e+01 3.717039e+01 3.717877e+01
## [351] 3.718156e+01 3.718436e+01 3.718994e+01 3.719832e+01 3.720112e+01
## [356] 3.720391e+01 3.720670e+01 3.720950e+01 3.721229e+01 3.721788e+01
## [361] 3.722067e+01 3.722346e+01 3.722626e+01 3.722905e+01 3.723184e+01
## [366] 3.723464e+01 3.723743e+01 3.724022e+01 3.724302e+01 3.724581e+01
## [371] 3.724860e+01 3.725140e+01 3.725419e+01 3.726257e+01 3.726816e+01
## [376] 3.727095e+01 3.727933e+01 3.728212e+01 3.728771e+01 3.729050e+01
## [381] 3.729609e+01 3.729888e+01 3.730168e+01 3.730447e+01 3.730726e+01
## [386] 3.731006e+01 3.731285e+01 3.732402e+01 3.733240e+01 3.733520e+01
## [391] 3.733799e+01 3.734078e+01 3.734358e+01 3.734637e+01 3.736034e+01
## [396] 3.736313e+01 3.736592e+01 3.737151e+01 3.737430e+01 3.737709e+01
## [401] 3.737989e+01 3.738268e+01 3.739106e+01 3.739385e+01 3.739665e+01
## [406] 3.739944e+01 3.740223e+01 3.740503e+01 3.740782e+01 3.741061e+01
## [411] 3.741620e+01 3.742179e+01 3.742458e+01 3.743296e+01 3.800000e+01
## [416] 3.800393e+01 3.800589e+01 3.801572e+01 3.801768e+01 3.802358e+01
## [421] 3.802554e+01 3.802750e+01 3.803143e+01 3.803340e+01 3.803536e+01
## [426] 3.804322e+01 3.804519e+01 3.804715e+01 3.805108e+01 3.805697e+01
## [431] 3.806090e+01 3.806483e+01 3.806680e+01 3.807269e+01 3.807662e+01
## [436] 3.808251e+01 3.808448e+01 3.808841e+01 3.809234e+01 3.809627e+01
## [441] 3.809823e+01 3.810216e+01 3.810806e+01 3.811198e+01 3.811395e+01
## [446] 3.812181e+01 3.812377e+01 3.812574e+01 3.812770e+01 3.812967e+01
## [451] 3.813163e+01 3.813360e+01 3.813949e+01 3.814145e+01 3.814538e+01
## [456] 3.815128e+01 3.900000e+01</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">gamma </code></pre></div>
<pre><code>## [1] 1.740065</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">gamma_p_value </code></pre></div>
<pre><code>## [1] 0.08184766</code></pre>
<p>There are two other regularly used metrics that include abundance measures. Note: we don’t have abundance measures for D-place data.</p>
<p><span class="math inline">\(IAC\)</span>, imbalance of abundance at the clade level, quantifies the relative deviation in the abundance distribution from a null case where individuals are evenly partitioned between clade splits. <span class="math inline">\(v\)</span> is the number of nodes in the phylogenetic tree. <span class="math inline">\(n_{i}\)</span> is, as defined above, the abundance of species <span class="math inline">\(i\)</span> in the assemblage. <span class="math inline">\(\eta_{k}\)</span> is the expected abundance species <span class="math inline">\(i\)</span> would have if the abundance was randomly split among lineages in the phylogenetic tree at each speciation event. is the number of lineages originating at node <span class="math inline">\(k\)</span> in the set <span class="math inline">\(s(k,root)\)</span>, which contains the nodes located on the path between node <span class="math inline">\(k\)</span> and the root of the phylogenetic tree. N is the total assemblage abundance <span class="citation">(Cadotte et al. <a href="#ref-Cadotte2010">2010</a>)</span>.</p>
<p><span class="math display">\[
\dfrac{\sum_{i=1}^{S}  |n_{i} - \hat{n_{i}}|}
{v} \\
\ \\
where \\ 
\ \\
\hat{n_{i}} = \dfrac{N}{\prod_{K \in s(i, root)}\eta_{k}}
\]</span></p>
<p><span class="math inline">\(I_{c}\)</span>, the Colless index, is the sum of the absolute differences in species richness between sister-clades at each internal node. For fully resolved trees, each internal node defines two sister-clades. <span class="math inline">\(S_{1k}\)</span> is the number of species descending from the first clade defined by node k and <span class="math inline">\(S_{2k}\)</span> that of the second clade. <span class="math inline">\(v\)</span> is, as defined above, the number of nodes in the phylogenetic <span class="citation">(Colless <a href="#ref-Colless1982">1982</a>)</span>.</p>
<p><span class="math display">\[
I_{c} = \sum_{k=1}^{v} |S_{1k} - S_{2k}|
\]</span></p>
</div>
<div id="macroevolutionary-rates" class="section level2">
<h2><span class="header-section-number">5.4</span> Macroevolutionary rates</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#function name = bd, function input = tree of type &#39;phylo&#39;</span>
bd &lt;-<span class="st"> </span><span class="cf">function</span> (tree) 
{
    tree<span class="op">$</span>edge.length &lt;-<span class="st"> </span>tree<span class="op">$</span>edge.length<span class="op">/</span><span class="kw">max</span>(tree<span class="op">$</span>edge.length)
    x &lt;-<span class="st"> </span><span class="kw">birthdeath</span>(tree)
    b &lt;-<span class="st"> </span>x<span class="op">$</span>para[<span class="dv">2</span>]<span class="op">/</span>(<span class="dv">1</span> <span class="op">-</span><span class="st"> </span>x<span class="op">$</span>para[<span class="dv">1</span>])
    d &lt;-<span class="st"> </span>b <span class="op">-</span><span class="st"> </span>x<span class="op">$</span>para[<span class="dv">2</span>]
    <span class="kw">c</span>(<span class="kw">setNames</span>(<span class="kw">c</span>(b, d), <span class="kw">c</span>(<span class="st">&quot;b&quot;</span>, <span class="st">&quot;d&quot;</span>)), x<span class="op">$</span>para)
}</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"> ## Speciation vs extinction rates and Net diversification
bds &lt;-<span class="st"> </span><span class="kw">bd</span>(this_tree)
speciation_rate &lt;-<span class="st"> </span>bds[<span class="dv">1</span>]
extinction_rate &lt;-<span class="st"> </span>bds[<span class="dv">2</span>]
extinction_per_speciation &lt;-<span class="st"> </span>bds[<span class="dv">3</span>]
speciation_minus_extinction &lt;-<span class="st"> </span>bds[<span class="dv">4</span>]</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">## Speciation vs extinction rates and Net diversification dependent on trait
<span class="co"># N.for.dom &lt;- table(this_world[, 6])</span>
<span class="co">#    if(length(N.for.dom) == 2) {</span>
par.div.dep &lt;-<span class="st"> </span><span class="kw">DivDep</span>( <span class="dt">mytree =</span> this_tree, <span class="dt">myWorld =</span> this_world)
trait_1_speciation &lt;-<span class="st"> </span>par.div.dep[<span class="dv">1</span>]
trait_2_speciation &lt;-<span class="st"> </span>par.div.dep[<span class="dv">2</span>]
trait_1_extinction &lt;-<span class="st"> </span>par.div.dep[<span class="dv">3</span>]
trait_2_extinction &lt;-<span class="st"> </span>par.div.dep[<span class="dv">4</span>]
transition_from_trait_1_to_<span class="dv">2</span> &lt;-<span class="st"> </span>par.div.dep[<span class="dv">5</span>]
transition_from_trait_2_to_<span class="dv">1</span> &lt;-<span class="st"> </span>par.div.dep[<span class="dv">6</span>]
transition_rate_ratio_1to2_over_2to1 &lt;-<span class="st"> </span>transition_from_trait_1_to_<span class="dv">2</span><span class="op">/</span>transition_from_trait_2_to_<span class="dv">1</span></code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">## Crown age per trait AUC and effect size
tip.length &lt;-<span class="st"> </span>this_tree<span class="op">$</span>edge.length[this_tree<span class="op">$</span>edge[, <span class="dv">2</span>] <span class="op">%in%</span><span class="st"> </span><span class="dv">1</span><span class="op">:</span><span class="kw">Ntip</span>(this_tree)]
tip.length &lt;-<span class="st"> </span>(tip.length <span class="op">-</span><span class="st"> </span><span class="kw">min</span>(tip.length)) <span class="op">/</span><span class="st"> </span>(<span class="kw">max</span>(tip.length) <span class="op">-</span><span class="st"> </span><span class="kw">min</span>(tip.length))
this_trait &lt;-<span class="st"> </span>this_world[<span class="kw">match</span>(this_tree<span class="op">$</span>tip.label, this_world[, <span class="dv">8</span>]), <span class="dv">6</span>]
tip.length.<span class="dv">2</span> &lt;-<span class="st"> </span>tip.length[this_trait <span class="op">==</span><span class="st"> </span><span class="dv">2</span>]
tip.length.<span class="dv">1</span> &lt;-<span class="st"> </span>tip.length[this_trait <span class="op">==</span><span class="st"> </span><span class="dv">1</span>]
model &lt;-<span class="st"> </span><span class="kw">glm</span>(<span class="kw">as.factor</span>(this_trait) <span class="op">~</span><span class="st"> </span><span class="kw">log</span>(tip.length <span class="op">+</span><span class="st"> </span><span class="dv">1</span>),
             <span class="dt">family =</span> <span class="st">&quot;binomial&quot;</span>)
effect.size &lt;-<span class="st"> </span>model<span class="op">$</span>coefficients[<span class="dv">2</span>]
<span class="co"># plot(y = this_trait - 1, x= log(tip.length))</span>
p &lt;-<span class="st"> </span><span class="kw">predict</span>(model, <span class="kw">as.factor</span>(this_trait), <span class="dt">type =</span> <span class="st">&quot;resp&quot;</span>)
<span class="co"># points(y = p, x = log(tip.length), col = &quot;red&quot;)</span>
pr &lt;-<span class="st"> </span><span class="kw">prediction</span>(p, <span class="kw">as.factor</span>(this_trait))
auc.model &lt;-<span class="st"> </span><span class="kw">performance</span>(pr, <span class="dt">measure =</span> <span class="st">&quot;auc&quot;</span>)<span class="op">@</span>y.values[[<span class="dv">1</span>]]</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">## Phylogenetic signal (D)
Phylogenetic_signal &lt;-<span class="st"> </span><span class="kw">Dsig</span>(<span class="dt">mytree =</span> this_tree, <span class="dt">myWorld =</span> this_world)</code></pre></div>
</div>
<div id="spatial-locations" class="section level2">
<h2><span class="header-section-number">5.5</span> Spatial Locations</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">## Spatial Analysis
nbs0 &lt;-<span class="st"> </span><span class="kw">knearneigh</span>(<span class="kw">as.matrix</span>(this_world[, <span class="dv">2</span><span class="op">:</span><span class="dv">3</span>]), <span class="dt">k =</span> <span class="dv">7</span>, <span class="dt">longlat =</span> <span class="ot">TRUE</span>)</code></pre></div>
<pre><code>## Warning in knearneigh(as.matrix(this_world[, 2:3]), k = 7, longlat = TRUE):
## knearneigh: identical points found</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">nbs &lt;-<span class="st"> </span><span class="kw">knn2nb</span>(nbs0, <span class="dt">sym =</span> <span class="ot">TRUE</span>) <span class="co"># 7 symmetric neighbors</span>
nbs.listw &lt;-<span class="st"> </span><span class="kw">nb2listw</span>(nbs)
factors.nbs &lt;-<span class="st"> </span><span class="kw">as.factor</span>(<span class="kw">ifelse</span>(<span class="kw">is.na</span>(this_world[, <span class="dv">6</span>]), <span class="dv">3</span>, this_world[, <span class="dv">6</span>]))
spatial.tests &lt;-<span class="st"> </span><span class="kw">joincount.test</span>(<span class="dt">fx =</span> factors.nbs, <span class="dt">listw =</span> nbs.listw)
spatial.tests.fora &lt;-<span class="st"> </span>spatial.tests[[<span class="dv">1</span>]]<span class="op">$</span>statistic
spatial.tests.dom &lt;-<span class="st"> </span>spatial.tests[[<span class="dv">2</span>]]<span class="op">$</span>statistic
<span class="co">#prevalence &lt;- (N.for.dom[1] - N.for.dom[2]) / sum(N.for.dom)</span></code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">results_summary_matrix_<span class="dv">1</span> &lt;-<span class="st"> </span><span class="kw">cbind</span>(

        number_of_branches,
        <span class="co">#Pylo_diversity_is_sum_of_BL,</span>
        <span class="co">#average_phylogenetic_diversity_is_mean_of_BL,</span>
        <span class="co">#variance_Pylo_diversity_is_variance_of_BL,</span>

        F_quadratic_entropy_is_sum_of_PD,
        Mean_pairwise_distance,
        variance_pairwise_distance,

        <span class="co">#Evolutionary_distinctiveness_sum,</span>
        <span class="co">#mean_Phylogenetic_isolation,</span>
        <span class="co">#variance_Phylogenetic_isolation,</span>

        gamma,
        gamma_p_value,
        speciation_rate,
        extinction_rate,
        extinction_per_speciation,
        speciation_minus_extinction,
        trait_1_speciation,
        trait_2_speciation ,
        trait_1_extinction ,
        trait_2_extinction ,
        transition_from_trait_1_to_<span class="dv">2</span> ,
        transition_from_trait_2_to_<span class="dv">1</span> ,
        transition_rate_ratio_1to2_over_2to1 ,
        Phylogenetic_signal,
        spatial.tests.fora,
        spatial.tests.dom,
       <span class="co"># prevalence,</span>
       <span class="co"># auc.model,</span>
        effect.size
      )
      <span class="co">#rownames(results_summary_matrix_1) &lt;- 1</span>

      <span class="co">#results_summary_matrix_2 &lt;- cbind(</span>
      <span class="co">#  c(Evolutionary_distinctiveness,NA),</span>
      <span class="co">#  lineages_through_time,</span>
      <span class="co">#  time_steps</span>
      <span class="co">#)</span>
      <span class="co">#colnames(results_summary_matrix_2) &lt;- c(&quot;Evolutionary_distinctiveness&quot;, &quot;lineages_through_time&quot;, &quot;time_steps&quot;)</span>
      <span class="co">#head(results_summary_matrix_2)</span>

      ### Returns from function in list form
      <span class="co">#returns &lt;- list(</span>
        <span class="co">#Branch_Lengths,</span>
        <span class="co">#Pairwise_dist,</span>
      <span class="co">#  results_summary_matrix_1,</span>
      <span class="co">#  results_summary_matrix_2</span>

      <span class="co">#)</span>

      <span class="co">#names(returns) &lt;- c(</span>
        <span class="co">#&quot;Branch_Lengths&quot;,</span>
        <span class="co">#&quot;Pairwise_distance&quot;,</span>
       <span class="co"># &quot;results_summary_of_single_value_outputs&quot;,</span>
       <span class="co"># &quot;results_summary_matrix_of_multi_value_outputs&quot;</span>
      <span class="co">#)</span></code></pre></div>
</div>
<div id="module2-returns-these-two-matrices-as-a-list" class="section level2">
<h2><span class="header-section-number">5.6</span> Module2() returns these two matrices as a list</h2>
<div id="here-is-the-exact-version-in-r" class="section level3">
<h3><span class="header-section-number">5.6.1</span> Here is the exact version in R</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">## This module analyzes the results from module 1 and returns a list based on how many values each stat returns
## Ty Tuff and Bruno Vilela
## 24 August 2016

###### Specify function ##############################

Module_<span class="dv">2</span> &lt;-<span class="st"> </span><span class="cf">function</span>(Module_1_output) {
  <span class="kw">cat</span>(<span class="st">&quot;</span><span class="ch">\n</span><span class="st">Analyzing: 0% [&quot;</span>)
  <span class="cf">if</span> (<span class="kw">any</span>(<span class="kw">is.na</span>(Module_1_output))) {
    <span class="kw">cat</span>(<span class="st">&quot;----------]&quot;</span>)
    <span class="kw">return</span>(<span class="ot">NA</span>)
  } <span class="cf">else</span> {

    this_tree &lt;-<span class="st"> </span>Module_1_output<span class="op">$</span>mytree
    this_world &lt;-<span class="st"> </span>Module_1_output<span class="op">$</span>myWorld

    ##### (0) Pull necessary variables from simulated trees and organize into a single object for all the tests below to pull from.

    <span class="co">#str(all_trees)</span>
    <span class="co">#str(this_tree)</span>

    ## 0a) Branch lengths
    Branch_Lengths &lt;-<span class="st"> </span>this_tree<span class="op">$</span>edge.length
    number_of_branches &lt;-<span class="st"> </span><span class="kw">length</span>(Branch_Lengths)

    <span class="co"># Anchor test = PD (Faith&#39;s phylogenetic diversity)</span>
    Pylo_diversity_is_sum_of_BL &lt;-<span class="st"> </span><span class="kw">sum</span>(Branch_Lengths)

    <span class="co"># avPD -- Average phylogenetic diversity</span>
    average_phylogenetic_diversity_is_mean_of_BL &lt;-<span class="st"> </span><span class="kw">mean</span>(Branch_Lengths)

    variance_Pylo_diversity_is_variance_of_BL &lt;-<span class="st"> </span><span class="kw">var</span>(Branch_Lengths)
    <span class="kw">cat</span>(<span class="st">&quot;-&quot;</span>)
    ## 0b) Pairwise distance between tips
    Pairwise_dist &lt;-<span class="st"> </span><span class="kw">cophenetic.phylo</span>(this_tree)
    <span class="kw">cat</span>(<span class="st">&quot;-&quot;</span>)
    <span class="co"># 2b) Pairwise distance -- Sum of pairwise distances</span>

    <span class="co"># F -- Extensive quadratic entropy</span>
    F_quadratic_entropy_is_sum_of_PD &lt;-<span class="st"> </span><span class="kw">sum</span>(Pairwise_dist)

    <span class="co">#Mean inter-species distances</span>

    <span class="co"># Anchor test = MPD (mean pairwise distance)</span>

    Mean_pairwise_distance &lt;-<span class="st"> </span><span class="kw">mean</span>(Pairwise_dist)

    <span class="kw">cat</span>(<span class="st">&quot;-&quot;</span>)
    <span class="co">#Pairwise distance/all distances -- Variance of pairwise distances</span>

    <span class="co"># Anchor test = VPD (variation of pairwise distance)</span>

    variance_pairwise_distance &lt;-<span class="st"> </span><span class="kw">var</span>(<span class="kw">as.vector</span>(Pairwise_dist))

    ## 0c) Phylogenetic isolation

    <span class="co"># Using equal.splits method, faster computation</span>
    Evolutionary_distinctiveness &lt;-<span class="st"> </span><span class="kw">evol.distinct2</span>(this_tree, <span class="dt">type =</span> <span class="st">&quot;fair.proportion&quot;</span>)
    <span class="kw">cat</span>(<span class="st">&quot;-&quot;</span>)
    <span class="co"># ED - Summed evolutionary distinctiveness</span>

    Evolutionary_distinctiveness_sum &lt;-<span class="st"> </span><span class="kw">sum</span>(Evolutionary_distinctiveness)

    ## 3d) Phylogenetic isolation -- Mean of species evolutionary distinctiveness

    <span class="co"># mean(ED)</span>

    mean_Phylogenetic_isolation &lt;-<span class="st"> </span><span class="kw">mean</span>(Evolutionary_distinctiveness)

    ## 4d) Phylogenetic isolation -- Variance of species isolation metrics

    <span class="co">#var(ED)</span>

    variance_Phylogenetic_isolation &lt;-<span class="st"> </span><span class="kw">var</span>(Evolutionary_distinctiveness)
    <span class="kw">cat</span>(<span class="st">&quot;-&quot;</span>)

    ## Tree topology

    <span class="co">#Gamma index</span>

    ltts &lt;-<span class="st"> </span><span class="kw">ltt</span>(this_tree, <span class="dt">gamma =</span> <span class="ot">TRUE</span>, <span class="dt">plot =</span> <span class="ot">FALSE</span>)
    lineages_through_time &lt;-<span class="st"> </span><span class="kw">as.numeric</span>(ltts[[<span class="dv">1</span>]])
    time_steps &lt;-<span class="st"> </span><span class="kw">as.numeric</span>(ltts[[<span class="dv">2</span>]])
    gamma &lt;-<span class="st"> </span>ltts[[<span class="dv">3</span>]]
    gamma_p_value &lt;-<span class="st"> </span>ltts[[<span class="dv">4</span>]]
    <span class="kw">cat</span>(<span class="st">&quot;-&quot;</span>)

    colless_stat &lt;-<span class="st"> </span><span class="kw">colless</span>(<span class="kw">as.treeshape</span>(this_tree))
    sackin_index &lt;-<span class="st"> </span><span class="kw">sackin</span>(<span class="kw">as.treeshape</span>(this_tree))
    tree_shape_stat &lt;-<span class="st"> </span><span class="kw">shape.statistic</span>(<span class="kw">as.treeshape</span>(this_tree))

    ##### (5) Tree metric -- Macroevolutionary - Rate and rate changes ###############
    ##################################################

    ## Speciation vs extinction rates and Net diversification
    bds &lt;-<span class="st"> </span><span class="kw">bd</span>(this_tree)
    speciation_rate &lt;-<span class="st"> </span>bds[<span class="dv">1</span>]
    extinction_rate &lt;-<span class="st"> </span>bds[<span class="dv">2</span>]
    extinction_per_speciation &lt;-<span class="st"> </span>bds[<span class="dv">3</span>]
    speciation_minus_extinction &lt;-<span class="st"> </span>bds[<span class="dv">4</span>]
    <span class="kw">cat</span>(<span class="st">&quot;-&quot;</span>)

    ## Speciation vs extinction rates and Net diversification dependent on trait
    N.for.dom &lt;-<span class="st"> </span><span class="kw">table</span>(this_world[, <span class="dv">6</span>])
    <span class="cf">if</span>(<span class="kw">length</span>(N.for.dom) <span class="op">==</span><span class="st"> </span><span class="dv">2</span>) {
      par.div.dep &lt;-<span class="st"> </span><span class="kw">DivDep</span>( <span class="dt">mytree =</span> this_tree, <span class="dt">myWorld =</span> this_world)
      trait_1_speciation &lt;-<span class="st"> </span>par.div.dep[<span class="dv">1</span>]
      trait_2_speciation &lt;-<span class="st"> </span>par.div.dep[<span class="dv">2</span>]
      trait_1_extinction &lt;-<span class="st"> </span>par.div.dep[<span class="dv">3</span>]
      trait_2_extinction &lt;-<span class="st"> </span>par.div.dep[<span class="dv">4</span>]
      transition_from_trait_1_to_<span class="dv">2</span> &lt;-<span class="st"> </span>par.div.dep[<span class="dv">5</span>]
      transition_from_trait_2_to_<span class="dv">1</span> &lt;-<span class="st"> </span>par.div.dep[<span class="dv">6</span>]
      transition_rate_ratio_1to2_over_2to1 &lt;-<span class="st"> </span>transition_from_trait_1_to_<span class="dv">2</span><span class="op">/</span>transition_from_trait_2_to_<span class="dv">1</span>
      <span class="kw">cat</span>(<span class="st">&quot;-&quot;</span>)

      ## Phylogenetic signal (D)
      Phylogenetic_signal &lt;-<span class="st"> </span><span class="kw">Dsig</span>(<span class="dt">mytree =</span> this_tree, <span class="dt">myWorld =</span> this_world)
      <span class="kw">cat</span>(<span class="st">&quot;-&quot;</span>)

      ## Spatial Analysis
      nbs0 &lt;-<span class="st"> </span><span class="kw">knearneigh</span>(<span class="kw">as.matrix</span>(this_world[, <span class="dv">2</span><span class="op">:</span><span class="dv">3</span>]), <span class="dt">k =</span> <span class="dv">7</span>, <span class="dt">longlat =</span> <span class="ot">TRUE</span>)
      nbs &lt;-<span class="st"> </span><span class="kw">knn2nb</span>(nbs0, <span class="dt">sym =</span> <span class="ot">TRUE</span>) <span class="co"># 7 symmetric neighbors</span>
      nbs.listw &lt;-<span class="st"> </span><span class="kw">nb2listw</span>(nbs)
      factors.nbs &lt;-<span class="st"> </span><span class="kw">as.factor</span>(<span class="kw">ifelse</span>(<span class="kw">is.na</span>(this_world[, <span class="dv">6</span>]), <span class="dv">3</span>, this_world[, <span class="dv">6</span>]))
      spatial.tests &lt;-<span class="st"> </span><span class="kw">joincount.test</span>(<span class="dt">fx =</span> factors.nbs, <span class="dt">listw =</span> nbs.listw)
      spatial.tests.fora &lt;-<span class="st"> </span>spatial.tests[[<span class="dv">1</span>]]<span class="op">$</span>statistic
      spatial.tests.dom &lt;-<span class="st"> </span>spatial.tests[[<span class="dv">2</span>]]<span class="op">$</span>statistic
      prevalence &lt;-<span class="st"> </span>(N.for.dom[<span class="dv">1</span>] <span class="op">-</span><span class="st"> </span>N.for.dom[<span class="dv">2</span>]) <span class="op">/</span><span class="st"> </span><span class="kw">sum</span>(N.for.dom)
      <span class="kw">cat</span>(<span class="st">&quot;-&quot;</span>)
    } <span class="cf">else</span> {
      trait_1_speciation &lt;-<span class="st"> </span><span class="ot">NA</span>
      trait_2_speciation &lt;-<span class="st"> </span><span class="ot">NA</span>
      trait_1_extinction &lt;-<span class="st"> </span><span class="ot">NA</span>
      trait_2_extinction &lt;-<span class="st"> </span><span class="ot">NA</span>
      transition_from_trait_1_to_<span class="dv">2</span> &lt;-<span class="st"> </span><span class="ot">NA</span>
      transition_from_trait_2_to_<span class="dv">1</span> &lt;-<span class="st"> </span><span class="ot">NA</span>
      transition_rate_ratio_1to2_over_2to1 &lt;-<span class="st"> </span><span class="ot">NA</span>
      Phylogenetic_signal &lt;-<span class="st"> </span><span class="ot">NA</span>
      spatial.tests.fora &lt;-<span class="st"> </span><span class="ot">NA</span>
      spatial.tests.dom &lt;-<span class="st"> </span><span class="ot">NA</span>
      prevalence &lt;-<span class="st"> </span><span class="kw">ifelse</span>(<span class="kw">names</span>(<span class="kw">table</span>(this_world[, <span class="dv">6</span>])[<span class="dv">1</span>]) <span class="op">==</span><span class="st"> &quot;1&quot;</span>, <span class="dv">1</span>,
                           <span class="op">-</span><span class="dv">1</span>)
      <span class="kw">cat</span>(<span class="st">&quot;---&quot;</span>)

    }

    results_summary_matrix_<span class="dv">1</span> &lt;-<span class="st"> </span><span class="kw">cbind</span>(

      number_of_branches,
      Pylo_diversity_is_sum_of_BL,
      average_phylogenetic_diversity_is_mean_of_BL,
      variance_Pylo_diversity_is_variance_of_BL,

      F_quadratic_entropy_is_sum_of_PD,
      Mean_pairwise_distance,
      variance_pairwise_distance,

      colless_stat ,
      sackin_index ,
      tree_shape_stat,

      Evolutionary_distinctiveness_sum,
      mean_Phylogenetic_isolation,
      variance_Phylogenetic_isolation,

      gamma,
      gamma_p_value,
      speciation_rate,
      extinction_rate,
      extinction_per_speciation,
      speciation_minus_extinction,
      trait_1_speciation,
      trait_2_speciation ,
      trait_1_extinction ,
      trait_2_extinction ,
      transition_from_trait_1_to_<span class="dv">2</span> ,
      transition_from_trait_2_to_<span class="dv">1</span> ,
      transition_rate_ratio_1to2_over_2to1 ,
      Phylogenetic_signal,
      spatial.tests.fora,
      spatial.tests.dom,
      prevalence
    )
    <span class="kw">rownames</span>(results_summary_matrix_<span class="dv">1</span>) &lt;-<span class="st"> </span><span class="dv">1</span>

    results_summary_matrix_<span class="dv">2</span> &lt;-<span class="st"> </span><span class="kw">cbind</span>(
      <span class="kw">c</span>(Evolutionary_distinctiveness,<span class="ot">NA</span>),
      lineages_through_time,
      time_steps
    )
    <span class="kw">colnames</span>(results_summary_matrix_<span class="dv">2</span>) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;Evolutionary_distinctiveness&quot;</span>,
                                            <span class="st">&quot;lineages_through_time&quot;</span>, <span class="st">&quot;time_steps&quot;</span>)
    <span class="kw">head</span>(results_summary_matrix_<span class="dv">2</span>)

    ### Returns from function in list form
    returns &lt;-<span class="st"> </span><span class="kw">list</span>(
      <span class="co">#Branch_Lengths,</span>
      <span class="co">#Pairwise_dist,</span>
      results_summary_matrix_<span class="dv">1</span>,
      results_summary_matrix_<span class="dv">2</span>

    )

    <span class="kw">names</span>(returns) &lt;-<span class="st"> </span><span class="kw">c</span>(
      <span class="co">#&quot;Branch_Lengths&quot;,</span>
      <span class="co">#&quot;Pairwise_distance&quot;,</span>
      <span class="st">&quot;results_summary_of_single_value_outputs&quot;</span>,
      <span class="st">&quot;results_summary_matrix_of_multi_value_outputs&quot;</span>
    )
    <span class="kw">cat</span>(<span class="st">&quot;] 100%&quot;</span>)

    <span class="kw">return</span>(returns)

  }
}


<span class="co">#Module_2(myOut)</span></code></pre></div>
</div>
</div>
<div id="references" class="section level2">
<h2><span class="header-section-number">5.7</span> References</h2>

</div>
</div>
<h3>References</h3>
<div id="refs" class="references">
<div id="ref-Tucker2016">
<p>Tucker, Caroline M., Marc W. Cadotte, Silvia B. Carvalho, T. Jonathan Davies, Simon Ferrier, Susanne A. Fritz, Rich Grenyer, et al. 2016. “A guide to phylogenetic metrics for conservation, community ecology and macroecology.” <em>Biological Reviews</em>. doi:<a href="https://doi.org/10.1111/brv.12252">10.1111/brv.12252</a>.</p>
</div>
<div id="ref-Faith1992">
<p>Faith, Daniel P. 1992. “Conservation evaluation and phylogenetic diversity.” <em>Biological Conservation</em> 61: 1–10. <a href="http://www.sciencedirect.com/science/article/pii/0006320792912013" class="uri">http://www.sciencedirect.com/science/article/pii/0006320792912013</a>.</p>
</div>
<div id="ref-Barker2002">
<p>Barker, Gary M. 2002. “Phylogenetic diversity : a quantitative framework for measurement of priority and achievement in biodiversity conservation.” <em>Biological Journal of the Linnean Society</em> 76: 165–94. <a href="http://www.jstor.org/stable/3070944?seq=1{\#}page{\_}scan{\_}tab{\_}contents" class="uri">http://www.jstor.org/stable/3070944?seq=1{\#}page{\_}scan{\_}tab{\_}contents</a>.</p>
</div>
<div id="ref-Rosauer2009">
<p>Rosauer, Dan F, Shawn W Laffan, Michael D Crisp, and Stephen C Donnellan. 2009. “Phylogenetic endemism : a new approach for identifying geographical concentrations of evolutionary history.” <em>Molecular Ecology</em> 18: 4061–72. doi:<a href="https://doi.org/10.1111/j.1365-294X.2009.04311.x">10.1111/j.1365-294X.2009.04311.x</a>.</p>
</div>
<div id="ref-Vellend2011">
<p>Vellend, Mark, William K Cornwell, Karen Magnuson-ford, and Arne Ø Mooers. 2011. “Measuring phylogenetic biodiversity.” In <em>Biological Diversity: Frontiers in Measurement and Assessment</em>, 194–207. Oxford, UK: Oxford University Press. <a href="http://phylodiversity.net/wcornwell/Vellend{\_}etal{\_}2011{\_}bookchap.pdf" class="uri">http://phylodiversity.net/wcornwell/Vellend{\_}etal{\_}2011{\_}bookchap.pdf</a>.</p>
</div>
<div id="ref-Clarke2001">
<p>Clarke, KR, and RM Warwick. 2001. “A further biodiversity index applicable to species lists: variation in taxonomic distinctness.” <em>Marine Ecology Progress Series</em> 216: 265–78. <a href="http://researchrepository.murdoch.edu.au/id/eprint/23107/" class="uri">http://researchrepository.murdoch.edu.au/id/eprint/23107/</a>.</p>
</div>
<div id="ref-Izsak2000">
<p>Izsak, Janos, and Laszio Papp. 2000. “A link between ecological diversity indices and measure of biodiversity.” <em>Ecological Modelling</em> 130 (1-3): 151–56. <a href="http://www.sciencedirect.com/science/article/pii/S0304380000002039" class="uri">http://www.sciencedirect.com/science/article/pii/S0304380000002039</a>.</p>
</div>
<div id="ref-Clarke1998">
<p>Clarke, KR, and RM Warwick. 1998. “Quantifying structural redundancy in ecological communities.” <em>Oecologia</em> 113: 278–89. <a href="http://link.springer.com/article/10.1007/s004420050379" class="uri">http://link.springer.com/article/10.1007/s004420050379</a>.</p>
</div>
<div id="ref-Webb2002">
<p>Webb, Campbell O, David D Ackerly, Mark A Mcpeek, and Michael J Donoghue. 2002. “Phylogenies and Community Ecology.” <em>Annual Review of Ecology and Systematics</em> 33: 475–505. doi:<a href="https://doi.org/10.1146/annurev.ecolsys.33.010802.150448">10.1146/annurev.ecolsys.33.010802.150448</a>.</p>
</div>
<div id="ref-Webb2008">
<p>Webb, Campbell O, David D Ackerly, and Steven W Kembel. 2008. “Phylocom: software for the analysis of phylogenetic community structure and trait evolution.” <em>Bioinformatics</em> 24 (18): 2098–2100. doi:<a href="https://doi.org/10.1093/bioinformatics/btn358">10.1093/bioinformatics/btn358</a>.</p>
</div>
<div id="ref-Kembel2010">
<p>Kembel, Steven W, Peter D Cowan, Matthew R Helmus, William K Cornwell, Helene Morlon, David D Ackerly, Simon P Blomberg, and Campbell O Webb. 2010. “Picante : R tools for integrating phylogenies and ecology.” <em>Bioinformatics</em> 26 (11): 1463–4. doi:<a href="https://doi.org/10.1093/bioinformatics/btq166">10.1093/bioinformatics/btq166</a>.</p>
</div>
<div id="ref-Helmus2010">
<p>Helmus, Matthew R, Wendel (Bill) Keller, Michael J Paterson, Norman D Yan, Charles H Cannon, and James A Rusak. 2010. “Communities contain closely related species during ecosystem disturbance.” <em>Ecology Letters</em> 13: 162–74. doi:<a href="https://doi.org/10.1111/j.1461-0248.2009.01411.x">10.1111/j.1461-0248.2009.01411.x</a>.</p>
</div>
<div id="ref-Rao1982">
<p>Rao, Radhakrishna C. 1982. “Diversity and Dissimilarity coefficients: a unified approach.” <em>Theoretical Population Biology</em> 21: 24–43. <a href="http://www.sciencedirect.com/science/article/pii/0040580982900041" class="uri">http://www.sciencedirect.com/science/article/pii/0040580982900041</a>.</p>
</div>
<div id="ref-Pavoine2005">
<p>Pavoine, S Ã, S Ollier, and D Pontier. 2005. “Measuring diversity from dissimilarities with Rao’s quadratic entropy: Are any dissimilarities suitable?” <em>Theoretical Population Biology</em> 67 (4): 231–39. doi:<a href="https://doi.org/10.1016/j.tpb.2005.01.004">10.1016/j.tpb.2005.01.004</a>.</p>
</div>
<div id="ref-Hardy2007">
<p>Hardy, Olivier J, and Bruno Senterre. 2007. “Characterizing the phylogenetic structure of communities by an additive partitioning of phylogenetic diversity.” <em>Journal of Ecology</em> 95: 493–506. doi:<a href="https://doi.org/10.1111/j.1365-2745.2007.01222.x">10.1111/j.1365-2745.2007.01222.x</a>.</p>
</div>
<div id="ref-Villeger2008">
<p>Villeger, Sebastien, Norman WH Mason, and David Mouillot. 2008. “New multidimensional functional diversity indices for a multifaceted framework in functional ecology.” <em>Ecology</em> 89 (8): 2290–2301. <a href="http://onlinelibrary.wiley.com/doi/10.1890/07-1206.1/full" class="uri">http://onlinelibrary.wiley.com/doi/10.1890/07-1206.1/full</a>.</p>
</div>
<div id="ref-Dehling2014">
<p>Dehling, D Matthias, Susanne A Fritz, Till Töpfer, Martin Päckert, Patrizia Estler, Katrin Böhning-gaese, and Matthias Schleuning. 2014. “Functional and phylogenetic diversity and assemblage structure of frugivorous birds along an elevational gradient in the tropical Andes IBS special issue.” <em>Ecography</em> 37: 1047–55. doi:<a href="https://doi.org/10.1111/ecog.00623">10.1111/ecog.00623</a>.</p>
</div>
<div id="ref-Redding2003">
<p>Redding, David W. 2003. “Incorporating genetic distinctness and reserve occupancy into a conservation priorisation approach.” Master’s thesis, University of East Anglia, Norwich.</p>
</div>
<div id="ref-Isaac2007">
<p>Isaac, Nick J B, Samuel T Turvey, Ben Collen, Carly Waterman, and Jonathan E M Baillie. 2007. “Mammals on the EDGE : Conservation Priorities Based on Threat and Phylogeny.” <em>PLoS ONE</em> 2 (3): e296. doi:<a href="https://doi.org/10.1371/journal.pone.0000296">10.1371/journal.pone.0000296</a>.</p>
</div>
<div id="ref-Safi2013a">
<p>Safi, Kamran, Katrina Armour-marshall, Jonathan E M Baillie, and Nick J B Isaac. 2013. “Global Patterns of Evolutionary Distinct and Globally Endangered Amphibians and Mammals.” <em>PLoS ONE</em> 8 (5): e63582–e63582. doi:<a href="https://doi.org/10.1371/journal.pone.0063582">10.1371/journal.pone.0063582</a>.</p>
</div>
<div id="ref-Cadotte2010">
<p>Cadotte, Marc W., T. Jonathan Davies, James Regetz, Steven W Kembel, Elsa Cleland, and Todd H. Oakley. 2010. “Phylogenetic diversity metrics for ecological communities : integrating species richness, abundance and evolutionary history.” <em>Ecology Letters</em> 13: 96–105. doi:<a href="https://doi.org/10.1111/j.1461-0248.2009.01405.x">10.1111/j.1461-0248.2009.01405.x</a>.</p>
</div>
<div id="ref-Pybus2000">
<p>Pybus, Oliver G, and Paul H Harvey. 2000. “Testing macro-evolutionary models using incomplete molecular phylogenies.” <em>Proceedings of the Royal Society B</em> 267 (1459): 2267–72. doi:<a href="https://doi.org/10.1098/rspb.2000.1278">10.1098/rspb.2000.1278</a>.</p>
</div>
<div id="ref-Colless1982">
<p>Colless, Donald H. 1982. “Review of phylogenetics: the theory and practice of phylogenetic systematics.” <em>Systematic Zoology</em> 31 (1): 100–104. <a href="http://www.jstor.org/stable/2413420" class="uri">http://www.jstor.org/stable/2413420</a>.</p>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="module-1-simulation-to-produce-a-world-and-a-tree.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="calling-module-1-and-module-2.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"weibo": false,
"instapper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/04_Module_2.Rmd",
"text": "Edit"
},
"download": ["bookdown-demo.pdf", "bookdown-demo.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.bootcss.com/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:" && /^https?:/.test(script.src))
      script.src  = script.src.replace(/^https?:/, '');
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
